<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Functional Analysis of TargetDatabase | Functional analysis</title>
  <meta name="description" content="Tutorial for MGS functional analysis" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Functional Analysis of TargetDatabase | Functional analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Tutorial for MGS functional analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Functional Analysis of TargetDatabase | Functional analysis" />
  
  <meta name="twitter:description" content="Tutorial for MGS functional analysis" />
  

<meta name="author" content="Bangzhuo Tong" />


<meta name="date" content="2022-11-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="functional-analysis-of-kegg.html"/>
<link rel="next" href="msea.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Function standard analysis workflow</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html"><i class="fa fa-check"></i><b>2</b> Functional Analysis of Metacyc database</a><ul>
<li class="chapter" data-level="2.1" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#environment-setup"><i class="fa fa-check"></i><b>2.1</b> Environment setup</a></li>
<li class="chapter" data-level="2.2" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#pipeline-file-processing"><i class="fa fa-check"></i><b>2.2</b> Pipeline file processing</a><ul>
<li class="chapter" data-level="2.2.1" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#merge-sample-files"><i class="fa fa-check"></i><b>2.2.1</b> Merge sample files</a></li>
<li class="chapter" data-level="2.2.2" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#split-strain-info"><i class="fa fa-check"></i><b>2.2.2</b> Split strain info</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#data-loading"><i class="fa fa-check"></i><b>2.3</b> Data loading</a><ul>
<li class="chapter" data-level="2.3.1" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#read-metadata"><i class="fa fa-check"></i><b>2.3.1</b> Read metadata:</a></li>
<li class="chapter" data-level="2.3.2" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#read-metaphlan2-profile-table"><i class="fa fa-check"></i><b>2.3.2</b> Read metaphlan2 profile table:</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#data-preprocess"><i class="fa fa-check"></i><b>2.4</b> Data preprocess</a><ul>
<li class="chapter" data-level="2.4.1" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#transforming-data"><i class="fa fa-check"></i><b>2.4.1</b> Transforming data</a></li>
<li class="chapter" data-level="2.4.2" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#remove-unppaed-pathways"><i class="fa fa-check"></i><b>2.4.2</b> Remove unppaed pathways</a></li>
<li class="chapter" data-level="2.4.3" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#aggregate-low-abundance-data"><i class="fa fa-check"></i><b>2.4.3</b> Aggregate low abundance data</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#plot-beta-diversity"><i class="fa fa-check"></i><b>2.5</b> Plot Beta diversity</a></li>
<li class="chapter" data-level="2.6" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#permanova-test"><i class="fa fa-check"></i><b>2.6</b> Permanova test</a></li>
<li class="chapter" data-level="2.7" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#differential-analysisda"><i class="fa fa-check"></i><b>2.7</b> Differential analysis(DA)</a><ul>
<li class="chapter" data-level="2.7.1" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#filter-low-prevalence-pathway"><i class="fa fa-check"></i><b>2.7.1</b> Filter low prevalence pathway</a></li>
<li class="chapter" data-level="2.7.2" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#lefse"><i class="fa fa-check"></i><b>2.7.2</b> LefSE</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#bacteria-contribution-to-differential-pathways"><i class="fa fa-check"></i><b>2.8</b> Bacteria contribution to differential pathways</a><ul>
<li class="chapter" data-level="2.8.1" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#generate-input-from-contribution-barplot"><i class="fa fa-check"></i><b>2.8.1</b> Generate input from contribution barplot</a></li>
<li class="chapter" data-level="2.8.2" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#plot-bacteria-contribution"><i class="fa fa-check"></i><b>2.8.2</b> Plot Bacteria contribution</a></li>
</ul></li>
<li class="chapter" data-level="2.9" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#session-info"><i class="fa fa-check"></i><b>2.9</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html"><i class="fa fa-check"></i><b>3</b> Functional Analysis of KEGG</a><ul>
<li class="chapter" data-level="3.1" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#environment-setup-1"><i class="fa fa-check"></i><b>3.1</b> Environment setup</a></li>
<li class="chapter" data-level="3.2" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#pipeline-file-processing-kegg"><i class="fa fa-check"></i><b>3.2</b> Pipeline file processing (KEGG)</a><ul>
<li class="chapter" data-level="3.2.1" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#remove-description-in-kegg-map-profile"><i class="fa fa-check"></i><b>3.2.1</b> Remove description in KEGG map profile</a></li>
<li class="chapter" data-level="3.2.2" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#merge-kegg-map-profile-of-samples"><i class="fa fa-check"></i><b>3.2.2</b> Merge KEGG map profile of samples</a></li>
<li class="chapter" data-level="3.2.3" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#remove-tmp-tsv"><i class="fa fa-check"></i><b>3.2.3</b> Remove tmp tsv</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#data-loading-kegg"><i class="fa fa-check"></i><b>3.3</b> Data loading (KEGG)</a></li>
<li class="chapter" data-level="3.4" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#data-preprocessing-kegg"><i class="fa fa-check"></i><b>3.4</b> Data preprocessing (KEGG)</a><ul>
<li class="chapter" data-level="3.4.1" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#remove-0-abundance-maps"><i class="fa fa-check"></i><b>3.4.1</b> Remove 0 abundance maps</a></li>
<li class="chapter" data-level="3.4.2" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#transforming-data-kegg"><i class="fa fa-check"></i><b>3.4.2</b> Transforming data (KEGG)</a></li>
<li class="chapter" data-level="3.4.3" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#aggregate-low-abundance-data-1"><i class="fa fa-check"></i><b>3.4.3</b> Aggregate low abundance data</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#standard-analysis-kegg"><i class="fa fa-check"></i><b>3.5</b> Standard Analysis (KEGG)</a><ul>
<li class="chapter" data-level="3.5.1" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#compositional-barplot"><i class="fa fa-check"></i><b>3.5.1</b> Compositional barplot</a></li>
<li class="chapter" data-level="3.5.2" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#beta-diversity"><i class="fa fa-check"></i><b>3.5.2</b> Beta diversity</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#differential-analysis-kegg"><i class="fa fa-check"></i><b>3.6</b> Differential analysis (KEGG)</a><ul>
<li class="chapter" data-level="3.6.1" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#filter-low-prevalence-pathway-kegg"><i class="fa fa-check"></i><b>3.6.1</b> Filter low prevalence pathway (KEGG)</a></li>
<li class="chapter" data-level="3.6.2" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#da-with-logistic-regression"><i class="fa fa-check"></i><b>3.6.2</b> DA with Logistic Regression</a></li>
<li class="chapter" data-level="3.6.3" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#show-da-result-with-volcano-plot"><i class="fa fa-check"></i><b>3.6.3</b> Show DA result with volcano plot</a></li>
<li class="chapter" data-level="3.6.4" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#show-da-result-with-pathview"><i class="fa fa-check"></i><b>3.6.4</b> Show DA result with pathview</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#session-information-kegg"><i class="fa fa-check"></i><b>3.7</b> Session information (KEGG)</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html"><i class="fa fa-check"></i><b>4</b> Functional Analysis of TargetDatabase</a><ul>
<li class="chapter" data-level="4.1" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#environment-setup-2"><i class="fa fa-check"></i><b>4.1</b> Environment setup</a></li>
<li class="chapter" data-level="4.2" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#pipeline-file-processing-1"><i class="fa fa-check"></i><b>4.2</b> Pipeline file processing</a><ul>
<li class="chapter" data-level="4.2.1" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#merge-sample-files-target-db"><i class="fa fa-check"></i><b>4.2.1</b> Merge sample files (Target db)</a></li>
<li class="chapter" data-level="4.2.2" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#split-strain-info-target-db"><i class="fa fa-check"></i><b>4.2.2</b> Split strain info (Target db)</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#data-loading-target-db"><i class="fa fa-check"></i><b>4.3</b> Data loading (Target db)</a></li>
<li class="chapter" data-level="4.4" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#data-preprocessing-target-db"><i class="fa fa-check"></i><b>4.4</b> Data preprocessing (Target db)</a><ul>
<li class="chapter" data-level="4.4.1" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#transforming-data-target-db"><i class="fa fa-check"></i><b>4.4.1</b> Transforming data (Target db)</a></li>
<li class="chapter" data-level="4.4.2" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#remove-unppaed-pathways-target-db"><i class="fa fa-check"></i><b>4.4.2</b> Remove unppaed pathways (Target db)</a></li>
<li class="chapter" data-level="4.4.3" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#aggregate-low-abundance-data-2"><i class="fa fa-check"></i><b>4.4.3</b> Aggregate low abundance data</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#standard-analysis"><i class="fa fa-check"></i><b>4.5</b> Standard analysis</a><ul>
<li class="chapter" data-level="4.5.1" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#compositional-barplot-1"><i class="fa fa-check"></i><b>4.5.1</b> Compositional barplot</a></li>
<li class="chapter" data-level="4.5.2" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#beta-diversity-1"><i class="fa fa-check"></i><b>4.5.2</b> Beta diversity</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#differential-analysis-da"><i class="fa fa-check"></i><b>4.6</b> Differential analysis (DA)</a><ul>
<li class="chapter" data-level="4.6.1" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#filter-low-prevalence-pathway-da"><i class="fa fa-check"></i><b>4.6.1</b> Filter low prevalence pathway (DA)</a></li>
<li class="chapter" data-level="4.6.2" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#da-with-lm"><i class="fa fa-check"></i><b>4.6.2</b> DA with LM</a></li>
<li class="chapter" data-level="4.6.3" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#show-da-result-with-volcano-plot-1"><i class="fa fa-check"></i><b>4.6.3</b> Show DA result with volcano plot</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#session-info-1"><i class="fa fa-check"></i><b>4.7</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="msea.html"><a href="msea.html"><i class="fa fa-check"></i><b>5</b> MSEA</a><ul>
<li class="chapter" data-level="5.1" data-path="msea.html"><a href="msea.html#execute-msea-with-python-script-on-server"><i class="fa fa-check"></i><b>5.1</b> Execute MSEA with python script on server</a></li>
<li class="chapter" data-level="5.2" data-path="msea.html"><a href="msea.html#environment-setup-3"><i class="fa fa-check"></i><b>5.2</b> Environment setup</a></li>
<li class="chapter" data-level="5.3" data-path="msea.html"><a href="msea.html#read-in-msea-result"><i class="fa fa-check"></i><b>5.3</b> Read in MSEA result</a></li>
<li class="chapter" data-level="5.4" data-path="msea.html"><a href="msea.html#filter-msea-result"><i class="fa fa-check"></i><b>5.4</b> Filter MSEA result</a></li>
<li class="chapter" data-level="5.5" data-path="msea.html"><a href="msea.html#visualization"><i class="fa fa-check"></i><b>5.5</b> Visualization</a></li>
<li class="chapter" data-level="5.6" data-path="msea.html"><a href="msea.html#run-enrichr-analysis"><i class="fa fa-check"></i><b>5.6</b> Run enrichR analysis</a></li>
<li class="chapter" data-level="5.7" data-path="msea.html"><a href="msea.html#session-info-2"><i class="fa fa-check"></i><b>5.7</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="picrust2.html"><a href="picrust2.html"><i class="fa fa-check"></i><b>6</b> PICRUSt2</a><ul>
<li class="chapter" data-level="6.1" data-path="picrust2.html"><a href="picrust2.html#environment-setup-4"><i class="fa fa-check"></i><b>6.1</b> Environment Setup</a></li>
<li class="chapter" data-level="6.2" data-path="picrust2.html"><a href="picrust2.html#run-picrust2"><i class="fa fa-check"></i><b>6.2</b> Run PICRUSt2</a></li>
<li class="chapter" data-level="6.3" data-path="picrust2.html"><a href="picrust2.html#interpret-result-table-from-picrust2"><i class="fa fa-check"></i><b>6.3</b> Interpret result table from PICRUSt2</a><ul>
<li class="chapter" data-level="6.3.1" data-path="picrust2.html"><a href="picrust2.html#read-input"><i class="fa fa-check"></i><b>6.3.1</b> Read input</a></li>
<li class="chapter" data-level="6.3.2" data-path="picrust2.html"><a href="picrust2.html#data-preprocess-picrust2"><i class="fa fa-check"></i><b>6.3.2</b> Data Preprocess (Picrust2)</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="picrust2.html"><a href="picrust2.html#session-info-3"><i class="fa fa-check"></i><b>6.4</b> Session info</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Functional analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="functional-analysis-of-targetdatabase" class="section level1">
<h1><span class="header-section-number">Chapter 4</span> Functional Analysis of TargetDatabase</h1>
<p><em><strong>This demo guides you how to analyze the pathway profile annotated from humman2 against 7 target databases.</strong></em></p>
<p>Target database inclues 7 types:</p>
<ol style="list-style-type: decimal">
<li><p>acetate</p></li>
<li><p>butyrate</p></li>
<li><p>formate</p></li>
<li><p>propionate</p></li>
<li><p>card</p></li>
<li><p>cazy</p></li>
<li><p>vfdb</p></li>
</ol>
<div class="figure">
<img src="flowcharts/FlowChart_TargetDB.jpg" alt="" />
<p class="caption">FlowChart_KEGG</p>
</div>
<p><strong>The analysis of the 7 types of target databases are pretty much the same, here we use CARD database as an example.</strong></p>
<hr />
<div id="environment-setup-2" class="section level2">
<h2><span class="header-section-number">4.1</span> Environment setup</h2>
<hr />
</div>
<div id="pipeline-file-processing-1" class="section level2">
<h2><span class="header-section-number">4.2</span> Pipeline file processing</h2>
<p><em>As described in the introduction chapter, <strong>a cohort containing 8 MGS samples of 4 patients from 2 groups in MAFLD project</strong> would be used as demo data in this tutorial.</em></p>
<div id="merge-sample-files-target-db" class="section level3">
<h3><span class="header-section-number">4.2.1</span> Merge sample files (Target db)</h3>
<p>Here we need to merge all CARD data from 8 samples into one profile table with the merge script from Humann2 <strong>in Bash</strong>. But if you have merged the sample data already, please skip this chunk and jump to <a href="functional-analysis-of-targetdatabase.html#data-loading-target-db">Data loading</a>.</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode markdown"><code class="sourceCode markdown"><span id="cb56-1"><a href="functional-analysis-of-targetdatabase.html#cb56-1"></a></span>
<span id="cb56-2"><a href="functional-analysis-of-targetdatabase.html#cb56-2"></a>docker run -i --rm -u $(id -u):$(id -g) -v /share/projects/SOP/Functional_Analysis/Tongbangzhuo/Demodata/:/in harbor.xbiome.com/xbiome/environments/humann2:2.8.1-2b8c5c3 bash -c &quot;merge_metaphlan_tables.py /in/PipelineOutput/*/humann2_card/*_genefamilies.tsv &gt; /in/Card/merged_card_profile.tsv&quot;</span></code></pre></div>
</div>
<div id="split-strain-info-target-db" class="section level3">
<h3><span class="header-section-number">4.2.2</span> Split strain info (Target db)</h3>
<p>Output profile from Humann2 pipeline contains strain information, we need to split the profile file into two files <strong>in Bash</strong>. But if you have splitted the sample data already, please skip this chunk and jump to <a href="functional-analysis-of-targetdatabase.html#data-loading-target-db">Data loading</a>.</p>
<ul>
<li>unstratified profile file</li>
<li>stratified profile file</li>
</ul>
<div class="sourceCode" id="cb57"><pre class="sourceCode markdown"><code class="sourceCode markdown"><span id="cb57-1"><a href="functional-analysis-of-targetdatabase.html#cb57-1"></a></span>
<span id="cb57-2"><a href="functional-analysis-of-targetdatabase.html#cb57-2"></a>grep -v &quot;s_&quot; /share/projects/SOP/Functional_Analysis/Tongbangzhuo/Demodata/Card/merged_card_profile.tsv &gt;  /share/projects/SOP/Functional_Analysis/Tongbangzhuo/Demodata/Card/merged_card_profile_unstratified.tsv</span>
<span id="cb57-3"><a href="functional-analysis-of-targetdatabase.html#cb57-3"></a></span>
<span id="cb57-4"><a href="functional-analysis-of-targetdatabase.html#cb57-4"></a>grep -E &quot;s_|ID&quot; /share/projects/SOP/Functional_Analysis/Tongbangzhuo/Demodata/Metacyc/merged_metacyc_profile.tsv &gt;  /share/projects/SOP/Functional_Analysis/Tongbangzhuo/Demodata/Card/merged_card_profile_stratified.tsv</span></code></pre></div>
<hr />
</div>
</div>
<div id="data-loading-target-db" class="section level2">
<h2><span class="header-section-number">4.3</span> Data loading (Target db)</h2>
<p><em><strong>Since the format of pathway profile annotated in 7 target databases are almost identical, We use card data as an example.</strong></em></p>
<p>Same as kegg pathway data, we rescale data at the begining of analysis to reduce the differences of sequencing depth across all samples.</p>
<p><em><strong>Read in card data</strong></em></p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="functional-analysis-of-targetdatabase.html#cb58-1"></a><span class="co">## Read in merged card profile data</span></span>
<span id="cb58-2"><a href="functional-analysis-of-targetdatabase.html#cb58-2"></a></span>
<span id="cb58-3"><a href="functional-analysis-of-targetdatabase.html#cb58-3"></a>merged_card_profile &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&#39;/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Demodata/Card/merged_card_profile_unstratified.tsv&#39;</span>,</span>
<span id="cb58-4"><a href="functional-analysis-of-targetdatabase.html#cb58-4"></a>                                  <span class="dt">sep =</span> <span class="st">&#39;</span><span class="ch">\t</span><span class="st">&#39;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">check.names =</span> <span class="ot">FALSE</span>, <span class="dt">na.strings =</span> <span class="st">&#39;&#39;</span>,<span class="dt">comment.char =</span> <span class="st">&#39;&#39;</span>, <span class="dt">quote =</span> <span class="st">&#39;&#39;</span>)</span>
<span id="cb58-5"><a href="functional-analysis-of-targetdatabase.html#cb58-5"></a></span>
<span id="cb58-6"><a href="functional-analysis-of-targetdatabase.html#cb58-6"></a><span class="kw">colnames</span>(merged_card_profile) <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">str_remove_all</span>(., <span class="st">&#39;_.+&#39;</span>)</span>
<span id="cb58-7"><a href="functional-analysis-of-targetdatabase.html#cb58-7"></a></span>
<span id="cb58-8"><a href="functional-analysis-of-targetdatabase.html#cb58-8"></a><span class="co"># ## Rename entry name, keep ARO number</span></span>
<span id="cb58-9"><a href="functional-analysis-of-targetdatabase.html#cb58-9"></a><span class="co"># merged_card_profile$ID %&lt;&gt;% str_replace_all(., &#39;.+(ARO:[0-9]+)\\|(.+)_\\[.+\\]&#39;, &#39;\\1|\\2&#39;)</span></span>
<span id="cb58-10"><a href="functional-analysis-of-targetdatabase.html#cb58-10"></a><span class="co"># </span></span>
<span id="cb58-11"><a href="functional-analysis-of-targetdatabase.html#cb58-11"></a><span class="co"># head(merged_card_profile, n = 2)</span></span>
<span id="cb58-12"><a href="functional-analysis-of-targetdatabase.html#cb58-12"></a></span>
<span id="cb58-13"><a href="functional-analysis-of-targetdatabase.html#cb58-13"></a><span class="co">## Read in metadata (Group information)</span></span>
<span id="cb58-14"><a href="functional-analysis-of-targetdatabase.html#cb58-14"></a>metadata &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&#39;/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Demodata/metadata.xls&#39;</span>, <span class="dt">check.names =</span> <span class="ot">FALSE</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>)</span>
<span id="cb58-15"><a href="functional-analysis-of-targetdatabase.html#cb58-15"></a></span>
<span id="cb58-16"><a href="functional-analysis-of-targetdatabase.html#cb58-16"></a>metadata <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">column_to_rownames</span>(<span class="st">&#39;SeqID&#39;</span>)</span></code></pre></div>
<hr />
</div>
<div id="data-preprocessing-target-db" class="section level2">
<h2><span class="header-section-number">4.4</span> Data preprocessing (Target db)</h2>
<div id="transforming-data-target-db" class="section level3">
<h3><span class="header-section-number">4.4.1</span> Transforming data (Target db)</h3>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="functional-analysis-of-targetdatabase.html#cb59-1"></a>rescaled_merged_card_profile &lt;-<span class="st"> </span>merged_card_profile <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">column_to_rownames</span>(<span class="st">&#39;ID&#39;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">apply</span>(., <span class="dv">2</span>, <span class="cf">function</span>(x) x<span class="op">/</span><span class="kw">sum</span>(x)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>()</span>
<span id="cb59-2"><a href="functional-analysis-of-targetdatabase.html#cb59-2"></a></span>
<span id="cb59-3"><a href="functional-analysis-of-targetdatabase.html#cb59-3"></a><span class="kw">head</span>(rescaled_merged_card_profile, <span class="dt">n =</span> <span class="dv">2</span>)</span></code></pre></div>
<pre><code>##                                                                    16453        16455        16456        16457        16461     16464        16465        16466
## UNMAPPED                                                    9.988510e-01 9.984107e-01 9.991099e-01 9.994578e-01 9.989571e-01 0.9993301 9.992888e-01 9.990494e-01
## gb|AAA20116.1|ARO:3000180|tetA(P)_[Clostridium_perfringens] 2.266749e-06 1.249741e-06 9.247141e-07 1.215390e-07 4.040324e-07 0.0000000 6.341580e-08 3.858159e-06</code></pre>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="functional-analysis-of-targetdatabase.html#cb61-1"></a><span class="kw">dim</span>(rescaled_merged_card_profile)</span></code></pre></div>
<pre><code>## [1] 908   8</code></pre>
</div>
<div id="remove-unppaed-pathways-target-db" class="section level3">
<h3><span class="header-section-number">4.4.2</span> Remove unppaed pathways (Target db)</h3>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="functional-analysis-of-targetdatabase.html#cb63-1"></a><span class="co">## In this chunk, we remove the umapped row in rescaled profile table because we are not able to intepret UNMAPPED entry.</span></span>
<span id="cb63-2"><a href="functional-analysis-of-targetdatabase.html#cb63-2"></a></span>
<span id="cb63-3"><a href="functional-analysis-of-targetdatabase.html#cb63-3"></a>card_profile &lt;-<span class="st"> </span>rescaled_merged_card_profile[<span class="kw">grep</span>(<span class="st">&quot;UNMAPPED&quot;</span>, <span class="kw">rownames</span>(rescaled_merged_card_profile), <span class="dt">invert =</span> <span class="ot">TRUE</span>), ]</span>
<span id="cb63-4"><a href="functional-analysis-of-targetdatabase.html#cb63-4"></a></span>
<span id="cb63-5"><a href="functional-analysis-of-targetdatabase.html#cb63-5"></a><span class="kw">dim</span>(card_profile)</span></code></pre></div>
<pre><code>## [1] 907   8</code></pre>
</div>
<div id="aggregate-low-abundance-data-2" class="section level3">
<h3><span class="header-section-number">4.4.3</span> Aggregate low abundance data</h3>
<p>In this chunck, we aggregate low abundance features to one row.<br />
<em><strong>1e-12 is an empirical threshold fot filtering low abundance feature. According to published paper <a href="https://doi.org/10.1016/j.chom.2019.07.004">Obese Individuals with and without Type 2 Diabetes Show Different Gut Microbial Functional Capacity and Composition</a>, pathway with top 50% mean abundance and top 50% variance are left. But in MaAsLin2ï¼Œpathway with abundance less than 10-10 are filtered by default.</strong></em></p>
<p>Note: Run <a href="functional-analysis-of-targetdatabase.html#transforming-data-target-db">Transforming data</a> and <a href="functional-analysis-of-targetdatabase.html#remove-unppaed-pathways-target-db">Remove unmapped pathways</a> before running this chunk!</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="functional-analysis-of-targetdatabase.html#cb65-1"></a>Filtered_card_profile &lt;-<span class="st"> </span><span class="kw">aggregate_low_abundance</span>(<span class="dt">input_data =</span> card_profile,</span>
<span id="cb65-2"><a href="functional-analysis-of-targetdatabase.html#cb65-2"></a>                                                           <span class="dt">threshold =</span> <span class="fl">1e-12</span>) <span class="co">## threshold should be modified based on your on study</span></span>
<span id="cb65-3"><a href="functional-analysis-of-targetdatabase.html#cb65-3"></a></span>
<span id="cb65-4"><a href="functional-analysis-of-targetdatabase.html#cb65-4"></a><span class="kw">dim</span>(Filtered_card_profile)</span></code></pre></div>
<pre><code>## [1] 907   8</code></pre>
<hr />
</div>
</div>
<div id="standard-analysis" class="section level2">
<h2><span class="header-section-number">4.5</span> Standard analysis</h2>
<p><strong>Note: All chunks in <a href="functional-analysis-of-targetdatabase.html#data-preprocessing-target-db">Data preprocessing</a> should be excuted before doing analysis in this chunk.</strong></p>
<div id="compositional-barplot-1" class="section level3">
<h3><span class="header-section-number">4.5.1</span> Compositional barplot</h3>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="functional-analysis-of-targetdatabase.html#cb67-1"></a><span class="co">## In this chunk, we construct stacked pathway barplot to depict the pathway composition of samples, we use function plot_stacked_bar from xviz to plot.</span></span>
<span id="cb67-2"><a href="functional-analysis-of-targetdatabase.html#cb67-2"></a><span class="co">## In case there are too much entries, we use parameter &quot;collapse&quot; in plot_stacked_bar function to integrate entries whose abundance are below given threshold into &quot;Others&quot;.</span></span>
<span id="cb67-3"><a href="functional-analysis-of-targetdatabase.html#cb67-3"></a><span class="co">## Note: Adjust your graph size to show complete graph.</span></span>
<span id="cb67-4"><a href="functional-analysis-of-targetdatabase.html#cb67-4"></a></span>
<span id="cb67-5"><a href="functional-analysis-of-targetdatabase.html#cb67-5"></a>compositional_plt &lt;-<span class="st"> </span>xviz<span class="op">::</span><span class="kw">plot_stacked_bar</span>(<span class="dt">otu_table =</span> Filtered_card_profile <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">t</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>(),</span>
<span id="cb67-6"><a href="functional-analysis-of-targetdatabase.html#cb67-6"></a>                 <span class="dt">metadata =</span> metadata, <span class="dt">collapse =</span> <span class="fl">0.01</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">vjust =</span> <span class="dv">1</span>))</span>
<span id="cb67-7"><a href="functional-analysis-of-targetdatabase.html#cb67-7"></a></span>
<span id="cb67-8"><a href="functional-analysis-of-targetdatabase.html#cb67-8"></a>compositional_plt</span></code></pre></div>
<p><img src="Function-standard-analysis-workflow_files/figure-html/Compositional%20barplot-1.png" width="576" /></p>
</div>
<div id="beta-diversity-1" class="section level3">
<h3><span class="header-section-number">4.5.2</span> Beta diversity</h3>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="functional-analysis-of-targetdatabase.html#cb68-1"></a><span class="co">## In this chunk, we inherit the concept of Beta diversity of microbial taxa data and apply it to pathway data to explore the similarity between samples.</span></span>
<span id="cb68-2"><a href="functional-analysis-of-targetdatabase.html#cb68-2"></a></span>
<span id="cb68-3"><a href="functional-analysis-of-targetdatabase.html#cb68-3"></a><span class="co">## construct phyloseq for beta diversity analysis</span></span>
<span id="cb68-4"><a href="functional-analysis-of-targetdatabase.html#cb68-4"></a></span>
<span id="cb68-5"><a href="functional-analysis-of-targetdatabase.html#cb68-5"></a>tmp_phyloseq &lt;-<span class="st"> </span><span class="kw">phyloseq</span>(<span class="kw">otu_table</span>(Filtered_card_profile, <span class="dt">taxa_are_rows =</span> <span class="ot">TRUE</span>),</span>
<span id="cb68-6"><a href="functional-analysis-of-targetdatabase.html#cb68-6"></a>                         <span class="kw">sample_data</span>(metadata))</span>
<span id="cb68-7"><a href="functional-analysis-of-targetdatabase.html#cb68-7"></a></span>
<span id="cb68-8"><a href="functional-analysis-of-targetdatabase.html#cb68-8"></a></span>
<span id="cb68-9"><a href="functional-analysis-of-targetdatabase.html#cb68-9"></a><span class="co">## PCoA plot</span></span>
<span id="cb68-10"><a href="functional-analysis-of-targetdatabase.html#cb68-10"></a>PCOA_plot &lt;-<span class="st"> </span>xviz<span class="op">::</span><span class="kw">plot_beta_diversity</span>(<span class="dt">phyloseq =</span> tmp_phyloseq,</span>
<span id="cb68-11"><a href="functional-analysis-of-targetdatabase.html#cb68-11"></a>                                       <span class="dt">feature =</span> <span class="st">&#39;Group&#39;</span>,</span>
<span id="cb68-12"><a href="functional-analysis-of-targetdatabase.html#cb68-12"></a>                                       <span class="dt">method =</span> <span class="st">&#39;bray&#39;</span>,</span>
<span id="cb68-13"><a href="functional-analysis-of-targetdatabase.html#cb68-13"></a>                                       <span class="dt">label =</span> <span class="ot">TRUE</span>)</span>
<span id="cb68-14"><a href="functional-analysis-of-targetdatabase.html#cb68-14"></a><span class="kw">print</span>(PCOA_plot)</span></code></pre></div>
<p><img src="Function-standard-analysis-workflow_files/figure-html/Beta%20diversity-1.png" width="672" /></p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="functional-analysis-of-targetdatabase.html#cb69-1"></a><span class="co">## PERMANOVA test &amp; beta dispersion test</span></span>
<span id="cb69-2"><a href="functional-analysis-of-targetdatabase.html#cb69-2"></a><span class="co">## We use PERMANOVA test to check the differences of function composition among different groups. Additionally, we also take homogeneity of group variance into consideration.</span></span>
<span id="cb69-3"><a href="functional-analysis-of-targetdatabase.html#cb69-3"></a></span>
<span id="cb69-4"><a href="functional-analysis-of-targetdatabase.html#cb69-4"></a>dispersion_permanova_res &lt;-<span class="st"> </span><span class="kw">run_permanova_betadisp</span>(<span class="dt">physeq =</span> tmp_phyloseq,</span>
<span id="cb69-5"><a href="functional-analysis-of-targetdatabase.html#cb69-5"></a>                                                   <span class="dt">vars =</span> <span class="st">&#39;Group&#39;</span></span>
<span id="cb69-6"><a href="functional-analysis-of-targetdatabase.html#cb69-6"></a>                                                   )</span>
<span id="cb69-7"><a href="functional-analysis-of-targetdatabase.html#cb69-7"></a></span>
<span id="cb69-8"><a href="functional-analysis-of-targetdatabase.html#cb69-8"></a>dispersion_permanova_res</span></code></pre></div>
<pre><code>## $betadisp_res
##   variable p_value                       analysis
## 1    Group   0.273 beta_dispersion_permutation999
## 
## $permanova_res
##   variable p_value        R2                 analysis
## 1    Group   0.476 0.1373393 permanova_permutation999</code></pre>
<hr />
</div>
</div>
<div id="differential-analysis-da" class="section level2">
<h2><span class="header-section-number">4.6</span> Differential analysis (DA)</h2>
<div id="filter-low-prevalence-pathway-da" class="section level3">
<h3><span class="header-section-number">4.6.1</span> Filter low prevalence pathway (DA)</h3>
<p>Low prevalence pathways are pathways only occur in minor samples.
In this chunk, we would remove pathways apperaing in less than max(2 , 5% of samples) from data set before doing analysis. <strong>Remember to run <a href="functional-analysis-of-targetdatabase.html#data-preprocessing-target-db">Data preprocessing</a> before running this chunk!</strong></p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="functional-analysis-of-targetdatabase.html#cb71-1"></a><span class="kw">print</span>(<span class="kw">paste0</span>(<span class="kw">nrow</span>(Filtered_card_profile), <span class="st">&#39; entries before filtering low prevalence data&#39;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;907 entries before filtering low prevalence data&quot;</code></pre>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="functional-analysis-of-targetdatabase.html#cb73-1"></a>Filtered_card_profile &lt;-<span class="st"> </span><span class="kw">filter_prevalence</span>(<span class="dt">otu_table =</span> Filtered_card_profile,</span>
<span id="cb73-2"><a href="functional-analysis-of-targetdatabase.html#cb73-2"></a>                  <span class="dt">threshold =</span> <span class="fl">0.05</span>,</span>
<span id="cb73-3"><a href="functional-analysis-of-targetdatabase.html#cb73-3"></a>                  <span class="dt">taxa_are_rows =</span> <span class="ot">TRUE</span></span>
<span id="cb73-4"><a href="functional-analysis-of-targetdatabase.html#cb73-4"></a>                  )</span>
<span id="cb73-5"><a href="functional-analysis-of-targetdatabase.html#cb73-5"></a></span>
<span id="cb73-6"><a href="functional-analysis-of-targetdatabase.html#cb73-6"></a><span class="kw">head</span>(Filtered_card_profile, <span class="dt">n =</span><span class="dv">2</span>)</span></code></pre></div>
<pre><code>##                                                                    16453        16455        16456       16457        16461        16464        16465        16466
## gb|AAA20116.1|ARO:3000180|tetA(P)_[Clostridium_perfringens] 2.266749e-06 1.249741e-06 9.247141e-07 1.21539e-07 4.040324e-07 0.000000e+00 6.341580e-08 3.858159e-06
## gb|AAA20117.1|ARO:3000195|tetB(P)_[Clostridium_perfringens] 4.260839e-06 3.980291e-06 1.026959e-06 0.00000e+00 4.795510e-07 1.736386e-07 4.890361e-07 9.308673e-06</code></pre>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="functional-analysis-of-targetdatabase.html#cb75-1"></a><span class="kw">print</span>(<span class="kw">paste0</span>(<span class="kw">nrow</span>(Filtered_card_profile), <span class="st">&#39; entries After filtering low prevalence data&#39;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;725 entries After filtering low prevalence data&quot;</code></pre>
</div>
<div id="da-with-lm" class="section level3">
<h3><span class="header-section-number">4.6.2</span> DA with LM</h3>
<p><strong>Note: Please <a href="functional-analysis-of-targetdatabase.html#filter-low-prevalence-pathway-da">Filter low prevalence pathway</a> before DA.</strong></p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="functional-analysis-of-targetdatabase.html#cb77-1"></a><span class="co">## In this chunk, you would be using logistic regression model to find pathways that are significantly enriched in certain group.</span></span>
<span id="cb77-2"><a href="functional-analysis-of-targetdatabase.html#cb77-2"></a></span>
<span id="cb77-3"><a href="functional-analysis-of-targetdatabase.html#cb77-3"></a><span class="co">## Due to the nature of compositional data, we cannot apply linear models to compositional data directly. </span></span>
<span id="cb77-4"><a href="functional-analysis-of-targetdatabase.html#cb77-4"></a><span class="co">## Transformation of relative abundance data should be carried out before feeding the data to LM.</span></span>
<span id="cb77-5"><a href="functional-analysis-of-targetdatabase.html#cb77-5"></a><span class="co">## Here, we add a very small value (1e-12) to the pathway profile table to avoid genrating NA during transformation, then use logit transformation to transform data. And eventually we apply LM to the transformed data</span></span>
<span id="cb77-6"><a href="functional-analysis-of-targetdatabase.html#cb77-6"></a></span>
<span id="cb77-7"><a href="functional-analysis-of-targetdatabase.html#cb77-7"></a><span class="co">## Adding small value to the profile table (The value is arbitrary).</span></span>
<span id="cb77-8"><a href="functional-analysis-of-targetdatabase.html#cb77-8"></a></span>
<span id="cb77-9"><a href="functional-analysis-of-targetdatabase.html#cb77-9"></a>DA_card_profile &lt;-<span class="st"> </span>Filtered_card_profile <span class="op">+</span><span class="st"> </span><span class="fl">1e-12</span></span>
<span id="cb77-10"><a href="functional-analysis-of-targetdatabase.html#cb77-10"></a></span>
<span id="cb77-11"><a href="functional-analysis-of-targetdatabase.html#cb77-11"></a><span class="co">## Reshape profile data table and use logit transformation.</span></span>
<span id="cb77-12"><a href="functional-analysis-of-targetdatabase.html#cb77-12"></a></span>
<span id="cb77-13"><a href="functional-analysis-of-targetdatabase.html#cb77-13"></a>DA_card_profile &lt;-<span class="st"> </span>DA_card_profile <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">t</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">rownames_to_column</span>(<span class="st">&#39;SeqID&#39;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>()</span>
<span id="cb77-14"><a href="functional-analysis-of-targetdatabase.html#cb77-14"></a>DA_metadata &lt;-<span class="st"> </span>metadata <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">rownames_to_column</span>(<span class="st">&#39;SeqID&#39;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>()</span>
<span id="cb77-15"><a href="functional-analysis-of-targetdatabase.html#cb77-15"></a></span>
<span id="cb77-16"><a href="functional-analysis-of-targetdatabase.html#cb77-16"></a><span class="co">## Reshape dataframe into long table</span></span>
<span id="cb77-17"><a href="functional-analysis-of-targetdatabase.html#cb77-17"></a>DA_input &lt;-<span class="st"> </span><span class="kw">merge</span>(DA_card_profile, DA_metadata, <span class="dt">by=</span><span class="st">&#39;SeqID&#39;</span>) <span class="op">%&gt;%</span><span class="st"> </span>reshape2<span class="op">::</span><span class="kw">melt</span>(<span class="dt">value.name =</span> <span class="st">&#39;RA&#39;</span>,</span>
<span id="cb77-18"><a href="functional-analysis-of-targetdatabase.html#cb77-18"></a>                                                                                         <span class="dt">variable.name =</span> <span class="st">&#39;PathwayIDs&#39;</span>)</span>
<span id="cb77-19"><a href="functional-analysis-of-targetdatabase.html#cb77-19"></a></span>
<span id="cb77-20"><a href="functional-analysis-of-targetdatabase.html#cb77-20"></a><span class="co">## Logit transformation</span></span>
<span id="cb77-21"><a href="functional-analysis-of-targetdatabase.html#cb77-21"></a>DA_input <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">RA_logit =</span> <span class="kw">log</span>(RA<span class="op">/</span>(<span class="dv">1</span><span class="op">-</span>RA)))</span>
<span id="cb77-22"><a href="functional-analysis-of-targetdatabase.html#cb77-22"></a></span>
<span id="cb77-23"><a href="functional-analysis-of-targetdatabase.html#cb77-23"></a><span class="co">## Fit data to LM</span></span>
<span id="cb77-24"><a href="functional-analysis-of-targetdatabase.html#cb77-24"></a><span class="co">## Loop over each pathway in two groups</span></span>
<span id="cb77-25"><a href="functional-analysis-of-targetdatabase.html#cb77-25"></a>LM_res &lt;-<span class="st"> </span>DA_input <span class="op">%&gt;%</span><span class="st">  </span><span class="kw">split</span>(.<span class="op">$</span>PathwayIDs) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">lapply</span>(., <span class="cf">function</span>(x){</span>
<span id="cb77-26"><a href="functional-analysis-of-targetdatabase.html#cb77-26"></a>  gml_res_summary &lt;-<span class="st"> </span><span class="kw">lm</span>(<span class="dt">data =</span> x, <span class="dt">formula =</span> RA_logit <span class="op">~</span><span class="st"> </span>Group) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">summary</span>() <span class="op">%&gt;%</span><span class="st"> </span>.<span class="op">$</span>coefficients <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.matrix</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">rownames_to_column</span>(<span class="dt">var =</span> <span class="st">&quot;Factors&quot;</span>)</span>
<span id="cb77-27"><a href="functional-analysis-of-targetdatabase.html#cb77-27"></a>  }</span>
<span id="cb77-28"><a href="functional-analysis-of-targetdatabase.html#cb77-28"></a>)</span>
<span id="cb77-29"><a href="functional-analysis-of-targetdatabase.html#cb77-29"></a></span>
<span id="cb77-30"><a href="functional-analysis-of-targetdatabase.html#cb77-30"></a><span class="co">## Merge all result in one table</span></span>
<span id="cb77-31"><a href="functional-analysis-of-targetdatabase.html#cb77-31"></a>LM_res &lt;-<span class="st"> </span>LM_res <span class="op">%&gt;%</span><span class="st"> </span>data.table<span class="op">::</span><span class="kw">rbindlist</span>(<span class="dt">idcol =</span> <span class="st">&quot;PathwayID&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(Factors <span class="op">!=</span><span class="st"> &quot;(Intercept)&quot;</span>)</span>
<span id="cb77-32"><a href="functional-analysis-of-targetdatabase.html#cb77-32"></a></span>
<span id="cb77-33"><a href="functional-analysis-of-targetdatabase.html#cb77-33"></a><span class="co">## Adjust p value using p.adjust function from stats package, you could choose different adjust method.</span></span>
<span id="cb77-34"><a href="functional-analysis-of-targetdatabase.html#cb77-34"></a>LM_res <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">adjust.p  =</span> stats<span class="op">::</span><span class="kw">p.adjust</span>(.<span class="op">$</span><span class="st">`</span><span class="dt">Pr(&gt;|t|)</span><span class="st">`</span>,</span>
<span id="cb77-35"><a href="functional-analysis-of-targetdatabase.html#cb77-35"></a>                                               <span class="dt">method =</span> <span class="st">&#39;BH&#39;</span>))</span>
<span id="cb77-36"><a href="functional-analysis-of-targetdatabase.html#cb77-36"></a></span>
<span id="cb77-37"><a href="functional-analysis-of-targetdatabase.html#cb77-37"></a><span class="co">## Calculate effect size (Odds ratio) of each feature</span></span>
<span id="cb77-38"><a href="functional-analysis-of-targetdatabase.html#cb77-38"></a>LM_res <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">OR =</span> <span class="kw">exp</span>(Estimate)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>()</span>
<span id="cb77-39"><a href="functional-analysis-of-targetdatabase.html#cb77-39"></a></span>
<span id="cb77-40"><a href="functional-analysis-of-targetdatabase.html#cb77-40"></a><span class="kw">head</span>(LM_res)</span></code></pre></div>
<pre><code>##                                                                                             PathwayID       Factors    Estimate Std. Error     t value   Pr(&gt;|t|)  adjust.p
## 1                                         gb|AAA20116.1|ARO:3000180|tetA(P)_[Clostridium_perfringens] Groupè‚é…¶æ­£å¸¸ -3.44760133  3.4786887 -0.99106347 0.35992376 0.9993136
## 2                                         gb|AAA20117.1|ARO:3000195|tetB(P)_[Clostridium_perfringens] Groupè‚é…¶æ­£å¸¸  3.16439828  3.7211162  0.85038953 0.42773400 0.9993136
## 3 gb|AAA23018.1|ARO:3004454|Campylobacter_coli_chloramphenicol_acetyltransferase_[Campylobacter_coli] Groupè‚é…¶æ­£å¸¸  9.72826026  3.0303720  3.21025277 0.01836118 0.5283449
## 4                                               gb|AAA23033.2|ARO:3000190|tetO_[Campylobacter_jejuni] Groupè‚é…¶æ­£å¸¸ -0.02887667  0.3625844 -0.07964124 0.93911245 0.9993136
## 5                                               gb|AAA25053.1|ARO:3000884|TEM-12_[Klebsiella_oxytoca] Groupè‚é…¶æ­£å¸¸  0.41660909  3.5283271  0.11807553 0.90986150 0.9993136
## 6                                             gb|AAA25293.1|ARO:3000192|tetS_[Listeria_monocytogenes] Groupè‚é…¶æ­£å¸¸ -0.39339749  0.2546723 -1.54472018 0.17336596 0.7089938
##             OR
## 1 3.182188e-02
## 2 2.367449e+01
## 3 1.678532e+04
## 4 9.715363e-01
## 5 1.516809e+00
## 6 6.747605e-01</code></pre>
</div>
<div id="show-da-result-with-volcano-plot-1" class="section level3">
<h3><span class="header-section-number">4.6.3</span> Show DA result with volcano plot</h3>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="functional-analysis-of-targetdatabase.html#cb79-1"></a><span class="co">## Plot volcano plot to show effect size (x-axis) and p value (y-axis) of pathways.</span></span>
<span id="cb79-2"><a href="functional-analysis-of-targetdatabase.html#cb79-2"></a><span class="co">## Here we only tend to hightlight pathways that satisfy adjust.p &lt; 0.05 and (OR &lt; 0.2 | OR &gt; 0.5) at the same time.</span></span>
<span id="cb79-3"><a href="functional-analysis-of-targetdatabase.html#cb79-3"></a><span class="co">## You can nevertheless choose different threshold accroding to your own data.</span></span>
<span id="cb79-4"><a href="functional-analysis-of-targetdatabase.html#cb79-4"></a></span>
<span id="cb79-5"><a href="functional-analysis-of-targetdatabase.html#cb79-5"></a>volcano_plot &lt;-<span class="st"> </span>LM_res <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">p.log =</span> <span class="op">-</span><span class="kw">log10</span>(<span class="st">`</span><span class="dt">Pr(&gt;|t|)</span><span class="st">`</span>), <span class="dt">log10OR =</span> <span class="kw">log10</span>(OR)) <span class="op">%&gt;%</span></span>
<span id="cb79-6"><a href="functional-analysis-of-targetdatabase.html#cb79-6"></a><span class="st">    </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> log10OR, <span class="dt">y =</span> p.log)) <span class="op">+</span></span>
<span id="cb79-7"><a href="functional-analysis-of-targetdatabase.html#cb79-7"></a><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="fl">0.5</span>) <span class="op">+</span></span>
<span id="cb79-8"><a href="functional-analysis-of-targetdatabase.html#cb79-8"></a><span class="st">    </span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="fl">0.5</span>, <span class="dt">color =</span> <span class="st">&quot;red&quot;</span>, <span class="dt">data =</span> . <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="st">`</span><span class="dt">Pr(&gt;|t|)</span><span class="st">`</span> <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span>(OR <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.2</span> <span class="op">|</span><span class="st"> </span>OR <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>))) <span class="op">+</span><span class="st"> </span></span>
<span id="cb79-9"><a href="functional-analysis-of-targetdatabase.html#cb79-9"></a><span class="st">    </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb79-10"><a href="functional-analysis-of-targetdatabase.html#cb79-10"></a><span class="st">    </span><span class="kw">geom_vline</span>(<span class="dt">xintercept =</span> <span class="kw">log10</span>(<span class="kw">c</span>(<span class="fl">0.05</span>, <span class="fl">0.1</span>, <span class="fl">0.2</span>, <span class="fl">0.5</span>, <span class="dv">1</span>, <span class="dv">2</span>)), <span class="dt">size =</span> <span class="fl">0.05</span>, <span class="dt">color =</span> <span class="st">&quot;grey&quot;</span>) <span class="op">+</span></span>
<span id="cb79-11"><a href="functional-analysis-of-targetdatabase.html#cb79-11"></a><span class="st">    </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="op">-</span><span class="kw">log10</span>(<span class="kw">c</span>(<span class="fl">0.7</span>)), <span class="dt">size =</span> <span class="fl">0.05</span>, <span class="dt">color =</span> <span class="st">&quot;grey&quot;</span>) <span class="op">+</span></span>
<span id="cb79-12"><a href="functional-analysis-of-targetdatabase.html#cb79-12"></a><span class="st">    </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>,</span>
<span id="cb79-13"><a href="functional-analysis-of-targetdatabase.html#cb79-13"></a>          <span class="dt">panel.grid.minor =</span> <span class="kw">element_blank</span>(),</span>
<span id="cb79-14"><a href="functional-analysis-of-targetdatabase.html#cb79-14"></a>          <span class="dt">panel.grid.major.x =</span> <span class="kw">element_blank</span>()</span>
<span id="cb79-15"><a href="functional-analysis-of-targetdatabase.html#cb79-15"></a>          ) <span class="op">+</span></span>
<span id="cb79-16"><a href="functional-analysis-of-targetdatabase.html#cb79-16"></a><span class="st">    </span><span class="kw">labs</span>(<span class="dt">x =</span> <span class="st">&quot;Estimated Odds Ratio&quot;</span>, <span class="dt">y =</span> <span class="st">&quot;FDR p-values(-log10)&quot;</span>)</span>
<span id="cb79-17"><a href="functional-analysis-of-targetdatabase.html#cb79-17"></a></span>
<span id="cb79-18"><a href="functional-analysis-of-targetdatabase.html#cb79-18"></a><span class="kw">print</span>(volcano_plot)</span></code></pre></div>
<p><img src="Function-standard-analysis-workflow_files/figure-html/visualize%20DA%20result%20by%20volcano%20plot-1.png" width="672" /></p>
<hr />
</div>
</div>
<div id="session-info-1" class="section level2">
<h2><span class="header-section-number">4.7</span> Session info</h2>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="functional-analysis-of-targetdatabase.html#cb80-1"></a>devtools<span class="op">::</span><span class="kw">session_info</span>()</span></code></pre></div>
<pre><code>## â”€ Session info â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
##  setting  value
##  version  R version 3.6.3 (2020-02-29)
##  os       Ubuntu 16.04.7 LTS
##  system   x86_64, linux-gnu
##  ui       RStudio
##  language (EN)
##  collate  en_IN.UTF-8
##  ctype    en_IN.UTF-8
##  tz       Asia/Hong_Kong
##  date     2022-11-09
##  rstudio  1.1.419 (server)
##  pandoc   2.7.3 @ /usr/bin/ (via rmarkdown)
## 
## â”€ Packages â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
##  ! package              * version    date (UTC) lib source
##    abind                  1.4-5      2016-07-21 [1] CRAN (R 3.6.3)
##    ade4                   1.7-17     2021-06-17 [1] CRAN (R 3.6.3)
##    ALDEx2               * 1.18.0     2019-10-29 [1] Bioconductor
##    annotate               1.64.0     2019-10-29 [1] Bioconductor
##    AnnotationDbi        * 1.58.0     2022-04-26 [1] Bioconductor
##    ape                    5.5        2021-04-25 [1] CRAN (R 3.6.3)
##    assertthat             0.2.1      2019-03-21 [2] CRAN (R 3.6.3)
##    backports              1.4.1      2021-12-13 [1] CRAN (R 3.6.3)
##    base64enc              0.1-3      2015-07-28 [2] CRAN (R 3.6.3)
##    bayesm                 3.1-4      2019-10-15 [1] CRAN (R 3.6.3)
##    biglm                  0.9-2.1    2020-11-27 [1] CRAN (R 3.6.3)
##    Biobase              * 2.46.0     2019-10-29 [2] Bioconductor
##    BiocGenerics         * 0.32.0     2019-10-29 [2] Bioconductor
##    BiocParallel         * 1.20.1     2019-12-21 [2] Bioconductor
##    biomformat             1.14.0     2019-10-29 [1] Bioconductor
##    Biostrings             2.54.0     2019-10-29 [1] Bioconductor
##    bit                    4.0.4      2020-08-04 [1] CRAN (R 3.6.3)
##    bit64                  4.0.5      2020-08-30 [1] CRAN (R 3.6.3)
##    bitops                 1.0-7      2021-04-24 [1] CRAN (R 3.6.3)
##    blob                   1.2.2      2021-07-23 [1] CRAN (R 3.6.3)
##    bookdown               0.24       2021-09-02 [1] CRAN (R 3.6.3)
##    brio                   1.1.3      2021-11-30 [2] CRAN (R 3.6.3)
##    broom                  0.7.12     2022-01-28 [1] CRAN (R 3.6.3)
##    bslib                  0.3.1      2021-10-06 [1] CRAN (R 3.6.3)
##    cachem                 1.0.5      2021-05-15 [1] CRAN (R 3.6.3)
##    callr                  3.7.0      2021-04-20 [2] CRAN (R 3.6.3)
##    car                    3.0-12     2021-11-06 [1] CRAN (R 3.6.3)
##    carData                3.0-4      2020-05-22 [1] CRAN (R 3.6.3)
##    caTools                1.18.2     2021-03-28 [1] CRAN (R 3.6.3)
##    cellranger             1.1.0      2016-07-27 [1] CRAN (R 3.6.3)
##    checkmate              2.0.0      2020-02-06 [1] CRAN (R 3.6.3)
##    circlize             * 0.4.13     2021-06-09 [1] CRAN (R 3.6.3)
##    cli                    3.1.0      2021-10-27 [1] CRAN (R 3.6.3)
##    clue                   0.3-59     2021-04-16 [1] CRAN (R 3.6.3)
##    cluster                2.1.0      2019-06-19 [2] CRAN (R 3.6.3)
##    coda                   0.19-4     2020-09-30 [1] CRAN (R 3.6.3)
##    codetools              0.2-16     2018-12-24 [2] CRAN (R 3.6.3)
##    coin                   1.4-2      2021-10-08 [1] CRAN (R 3.6.3)
##    colorspace             2.0-2      2021-06-24 [1] CRAN (R 3.6.3)
##    ComplexHeatmap       * 2.2.0      2019-10-29 [1] Bioconductor
##    compositions           2.0-2      2021-07-14 [1] CRAN (R 3.6.3)
##    cowplot              * 1.1.1      2020-12-30 [1] CRAN (R 3.6.3)
##    crayon                 1.5.0      2022-02-14 [1] CRAN (R 3.6.3)
##    curl                   4.3.2      2021-06-23 [2] CRAN (R 3.6.3)
##    dada2                * 1.14.1     2020-02-22 [1] Bioconductor
##    data.table           * 1.14.0     2021-02-21 [1] CRAN (R 3.6.3)
##    DBI                    1.1.1      2021-01-15 [1] CRAN (R 3.6.3)
##    dbplyr                 2.1.1      2021-04-06 [1] CRAN (R 3.6.3)
##    DelayedArray         * 0.12.3     2020-04-09 [2] Bioconductor
##    DelayedMatrixStats     1.8.0      2019-10-29 [1] Bioconductor
##    DEoptimR               1.0-9      2021-05-24 [1] CRAN (R 3.6.3)
##    desc                   1.4.1      2022-03-06 [2] CRAN (R 3.6.3)
##    DESeq2               * 1.26.0     2019-10-29 [1] Bioconductor
##    devtools               2.4.3      2021-11-30 [1] CRAN (R 3.6.3)
##    digest                 0.6.29     2021-12-01 [1] CRAN (R 3.6.3)
##    dplyr                * 1.0.6      2021-05-05 [1] CRAN (R 3.6.3)
##    edgeR                  3.28.1     2020-02-26 [1] Bioconductor
##    ellipsis               0.3.2      2021-04-29 [1] CRAN (R 3.6.3)
##    EnhancedVolcano      * 1.4.0      2019-10-29 [1] Bioconductor
##    enrichR              * 3.0        2021-02-02 [1] CRAN (R 3.6.3)
##    evaluate               0.15       2022-02-18 [2] CRAN (R 3.6.3)
##    fansi                  1.0.2      2022-01-14 [1] CRAN (R 3.6.3)
##    farver                 2.1.0      2021-02-28 [2] CRAN (R 3.6.3)
##    fastmap                1.1.0      2021-01-25 [1] CRAN (R 3.6.3)
##    fdrtool                1.2.17     2021-11-13 [1] CRAN (R 3.6.3)
##    forcats              * 0.5.1      2021-01-27 [1] CRAN (R 3.6.3)
##    foreach                1.5.2      2022-02-02 [2] CRAN (R 3.6.3)
##    foreign                0.8-75     2020-01-20 [2] CRAN (R 3.6.3)
##    formatR                1.12       2022-03-31 [2] CRAN (R 3.6.3)
##    Formula                1.2-4      2020-10-16 [1] CRAN (R 3.6.3)
##    fs                     1.5.2      2021-12-08 [1] CRAN (R 3.6.3)
##    futile.logger          1.4.3      2016-07-10 [2] CRAN (R 3.6.3)
##    futile.options         1.0.1      2018-04-20 [2] CRAN (R 3.6.3)
##    genefilter             1.68.0     2019-10-29 [1] Bioconductor
##    geneplotter            1.64.0     2019-10-29 [1] Bioconductor
##    generics               0.1.2      2022-01-31 [1] CRAN (R 3.6.3)
##    GenomeInfoDb         * 1.22.1     2020-03-27 [2] Bioconductor
##    GenomeInfoDbData       1.2.2      2020-08-24 [2] Bioconductor
##    GenomicAlignments      1.22.1     2019-11-12 [1] Bioconductor
##    GenomicRanges        * 1.38.0     2019-10-29 [2] Bioconductor
##    getopt                 1.20.3     2019-03-22 [1] CRAN (R 3.6.3)
##    GetoptLong             1.0.5      2020-12-15 [1] CRAN (R 3.6.3)
##    GGally               * 2.1.2      2021-06-21 [1] CRAN (R 3.6.3)
##    ggbipart             * 0.1.2      2022-07-20 [1] Github (pedroj/bipartite_plots@162f577)
##    ggExtra              * 0.9        2019-08-27 [1] CRAN (R 3.6.3)
##    ggplot2              * 3.3.5      2021-06-25 [1] CRAN (R 3.6.3)
##    ggpubr               * 0.4.0      2020-06-27 [1] CRAN (R 3.6.3)
##    ggrepel              * 0.9.1      2021-01-15 [2] CRAN (R 3.6.3)
##    ggsci                * 2.9        2018-05-14 [1] CRAN (R 3.6.3)
##    ggsignif               0.6.3      2021-09-09 [1] CRAN (R 3.6.3)
##    glmnet                 4.1-2      2021-06-24 [1] CRAN (R 3.6.3)
##    GlobalOptions          0.1.2      2020-06-10 [1] CRAN (R 3.6.3)
##    glue                   1.6.1      2022-01-22 [1] CRAN (R 3.6.3)
##    GMPR                   0.1.3      2021-05-17 [1] local
##    gplots                 3.1.1      2020-11-28 [1] CRAN (R 3.6.3)
##    graph                  1.64.0     2019-10-29 [1] Bioconductor
##    gridExtra              2.3        2017-09-09 [2] CRAN (R 3.6.3)
##    gtable                 0.3.0      2019-03-25 [2] CRAN (R 3.6.3)
##    gtools                 3.9.2      2021-06-06 [1] CRAN (R 3.6.3)
##    haven                  2.4.1      2021-04-23 [1] CRAN (R 3.6.3)
##    highr                  0.9        2021-04-16 [1] CRAN (R 3.6.3)
##    Hmisc                  4.5-0      2021-02-28 [1] CRAN (R 3.6.3)
##    hms                    1.1.1      2021-09-26 [1] CRAN (R 3.6.3)
##    htmlTable              2.3.0      2021-10-12 [1] CRAN (R 3.6.3)
##    htmltools              0.5.2      2021-08-25 [1] CRAN (R 3.6.3)
##    htmlwidgets            1.5.4      2021-09-08 [2] CRAN (R 3.6.3)
##    httpuv                 1.6.1      2021-05-07 [1] CRAN (R 3.6.3)
##    httr                   1.4.3      2022-05-04 [2] CRAN (R 3.6.3)
##    hwriter                1.3.2      2014-09-10 [1] CRAN (R 3.6.3)
##    igraph                 1.3.1      2022-04-20 [2] CRAN (R 3.6.3)
##    IHW                    1.14.0     2019-10-29 [1] Bioconductor
##    IRanges              * 2.20.2     2020-01-13 [2] Bioconductor
##    iterators              1.0.14     2022-02-05 [2] CRAN (R 3.6.3)
##    jpeg                   0.1-9      2021-07-24 [1] CRAN (R 3.6.3)
##    jquerylib              0.1.4      2021-04-26 [1] CRAN (R 3.6.3)
##    jsonlite               1.8.0      2022-02-22 [2] CRAN (R 3.6.3)
##    KEGGgraph              1.46.0     2019-10-29 [1] Bioconductor
##    KEGGREST               1.26.1     2019-11-06 [1] Bioconductor
##    KernSmooth             2.23-16    2019-10-15 [2] CRAN (R 3.6.3)
##    knitr                  1.36       2021-09-29 [1] CRAN (R 3.6.3)
##    labeling               0.4.2      2020-10-20 [2] CRAN (R 3.6.3)
##    lambda.r               1.2.4      2019-09-18 [2] CRAN (R 3.6.3)
##    later                  1.3.0      2021-08-18 [2] CRAN (R 3.6.3)
##    lattice              * 0.20-38    2018-11-04 [2] CRAN (R 3.6.3)
##    latticeExtra           0.6-29     2019-12-19 [1] CRAN (R 3.6.3)
##    lazyeval               0.2.2      2019-03-15 [2] CRAN (R 3.6.3)
##    libcoin                1.0-9      2021-09-27 [1] CRAN (R 3.6.3)
##    lifecycle              1.0.1      2021-09-24 [1] CRAN (R 3.6.3)
##    limma                  3.42.2     2020-02-03 [2] Bioconductor
##    locfit                 1.5-9.4    2020-03-25 [1] CRAN (R 3.6.3)
##    lpsymphony             1.14.0     2019-10-29 [1] Bioconductor (R 3.6.3)
##    lubridate              1.7.10     2021-02-26 [1] CRAN (R 3.6.3)
##    Maaslin2               1.7.3      2022-03-23 [1] Github (biobakery/maaslin2@8d090e4)
##    magrittr             * 2.0.2      2022-01-26 [1] CRAN (R 3.6.3)
##    MASS                   7.3-54     2021-05-03 [1] CRAN (R 3.6.3)
##    Matrix                 1.3-4      2021-06-01 [1] CRAN (R 3.6.3)
##    matrixStats          * 0.60.0     2021-07-26 [1] CRAN (R 3.6.3)
##    mbzinb                 0.2        2021-06-23 [1] local
##    memoise                2.0.1      2021-11-26 [2] CRAN (R 3.6.3)
##    metagenomeSeq          1.28.2     2020-02-03 [1] Bioconductor
##    metamicrobiomeR        1.1        2021-02-03 [1] local
##    mgcv                   1.8-31     2019-11-09 [2] CRAN (R 3.6.3)
##    microbiome             1.8.0      2019-10-29 [1] Bioconductor
##    mime                   0.12       2021-09-28 [2] CRAN (R 3.6.3)
##    miniUI                 0.1.1.1    2018-05-18 [1] CRAN (R 3.6.3)
##    modelr                 0.1.8      2020-05-19 [1] CRAN (R 3.6.3)
##    modeltools             0.2-23     2020-03-05 [1] CRAN (R 3.6.3)
##    multcomp               1.4-17     2021-04-29 [1] CRAN (R 3.6.3)
##    multtest               2.42.0     2019-10-29 [2] Bioconductor
##    munsell                0.5.0      2018-06-12 [2] CRAN (R 3.6.3)
##    mvtnorm                1.1-3      2021-10-08 [1] CRAN (R 3.6.3)
##    network              * 1.17.1     2021-06-14 [1] CRAN (R 3.6.3)
##    nlme                   3.1-144    2020-02-06 [2] CRAN (R 3.6.3)
##    nnet                   7.3-12     2016-02-02 [2] CRAN (R 3.6.3)
##    optparse               1.7.1      2021-10-08 [1] CRAN (R 3.6.3)
##    org.Hs.eg.db         * 3.10.0     2021-12-08 [1] Bioconductor
##    pathview             * 1.26.0     2019-10-29 [1] Bioconductor
##    pcaPP                  1.9-74     2021-04-23 [1] CRAN (R 3.6.3)
##    permute              * 0.9-5      2019-03-12 [1] CRAN (R 3.6.3)
##    phyloseq             * 1.30.0     2019-10-29 [1] Bioconductor
##    pillar                 1.7.0      2022-02-01 [1] CRAN (R 3.6.3)
##    pkgbuild               1.3.1      2021-12-20 [2] CRAN (R 3.6.3)
##    pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 3.6.3)
##    pkgload                1.2.4      2021-11-30 [2] CRAN (R 3.6.3)
##    plotly               * 4.10.0     2021-10-09 [1] CRAN (R 3.6.3)
##    plyr                   1.8.7      2022-03-24 [2] CRAN (R 3.6.3)
##    png                    0.1-7      2013-12-03 [1] CRAN (R 3.6.3)
##    prettyunits            1.1.1      2020-01-24 [2] CRAN (R 3.6.3)
##    processx               3.5.3      2022-03-25 [2] CRAN (R 3.6.3)
##    promises               1.2.0.1    2021-02-11 [2] CRAN (R 3.6.3)
##    protoclust             1.6.3      2019-01-31 [1] CRAN (R 3.6.3)
##    ps                     1.7.0      2022-04-23 [2] CRAN (R 3.6.3)
##    pscl                   1.5.5      2020-03-07 [1] CRAN (R 3.6.3)
##    purrr                * 0.3.4      2020-04-17 [2] CRAN (R 3.6.3)
##    qvalue                 2.18.0     2019-10-29 [1] Bioconductor
##    R6                     2.5.1      2021-08-19 [1] CRAN (R 3.6.3)
##    RAIDA                  1.0        2021-06-23 [1] local
##    ranacapa               0.1.0      2021-06-18 [1] Github (gauravsk/ranacapa@58c0cab)
##    RColorBrewer         * 1.1-3      2022-04-03 [2] CRAN (R 3.6.3)
##    Rcpp                 * 1.0.7      2021-07-07 [1] CRAN (R 3.6.3)
##    RcppParallel           5.1.4      2021-05-04 [1] CRAN (R 3.6.3)
##    RCurl                  1.98-1.6   2022-02-08 [2] CRAN (R 3.6.3)
##    readr                * 2.0.0      2021-07-20 [1] CRAN (R 3.6.3)
##    readxl               * 1.3.1      2019-03-13 [1] CRAN (R 3.6.3)
##    remotes                2.4.2      2021-11-30 [1] CRAN (R 3.6.3)
##    reprex                 2.0.1      2021-08-05 [1] CRAN (R 3.6.3)
##    reshape                0.8.9      2022-04-12 [1] CRAN (R 3.6.3)
##    reshape2             * 1.4.4      2020-04-09 [2] CRAN (R 3.6.3)
##    Rgraphviz              2.30.0     2019-10-29 [1] Bioconductor
##    rhdf5                  2.30.1     2019-11-26 [1] Bioconductor
##    Rhdf5lib               1.8.0      2019-10-29 [1] Bioconductor
##    rJava                  1.0-5      2021-09-24 [1] CRAN (R 3.6.3)
##    rjson                  0.2.20     2018-06-08 [1] CRAN (R 3.6.3)
##  R rlang                  1.0.2      &lt;NA&gt;       [2] &lt;NA&gt;
##    rmarkdown              2.11       2021-09-14 [1] CRAN (R 3.6.3)
##    robustbase             0.93-9     2021-09-27 [1] CRAN (R 3.6.3)
##    rpart                  4.1-15     2019-04-12 [2] CRAN (R 3.6.3)
##    rprojroot              2.0.2      2020-11-15 [1] CRAN (R 3.6.3)
##    Rsamtools              2.2.3      2020-02-23 [1] Bioconductor
##    RSQLite                2.2.7      2021-04-22 [1] CRAN (R 3.6.3)
##    rstatix                0.7.0      2021-02-13 [1] CRAN (R 3.6.3)
##    rstudioapi             0.13       2020-11-12 [2] CRAN (R 3.6.3)
##    Rtsne                  0.15       2018-11-10 [1] CRAN (R 3.6.3)
##    rvest                  1.0.2      2021-10-16 [1] CRAN (R 3.6.3)
##    S4Vectors            * 0.24.4     2020-04-09 [2] Bioconductor
##    sandwich               3.0-1      2021-05-18 [1] CRAN (R 3.6.3)
##    sass                   0.4.0      2021-05-12 [1] CRAN (R 3.6.3)
##    scales                 1.2.0      2022-04-13 [2] CRAN (R 3.6.3)
##    seqinr               * 4.2-8      2021-06-09 [1] CRAN (R 3.6.3)
##    sessioninfo            1.2.2      2021-12-06 [2] CRAN (R 3.6.3)
##    shape                  1.4.6      2021-05-19 [1] CRAN (R 3.6.3)
##    shiny                  1.7.1      2021-10-02 [1] CRAN (R 3.6.3)
##    ShortRead              1.44.3     2020-02-03 [1] Bioconductor
##    slam                   0.1-49     2021-11-17 [1] CRAN (R 3.6.3)
##    statnet.common         4.5.0      2021-06-05 [1] CRAN (R 3.6.3)
##    stringi                1.7.4      2021-08-25 [1] CRAN (R 3.6.3)
##    stringr              * 1.4.0      2019-02-10 [2] CRAN (R 3.6.3)
##    SummarizedExperiment * 1.16.1     2019-12-19 [2] Bioconductor
##    survival               3.1-8      2019-12-03 [2] CRAN (R 3.6.3)
##    tensorA                0.36.2     2020-11-19 [1] CRAN (R 3.6.3)
##    testthat               3.1.4      2022-04-26 [2] CRAN (R 3.6.3)
##    textshape              1.7.3      2021-05-28 [1] CRAN (R 3.6.3)
##    TH.data                1.1-0      2021-09-27 [1] CRAN (R 3.6.3)
##    tibble               * 3.1.6      2021-11-07 [1] CRAN (R 3.6.3)
##    tidyr                * 1.2.0      2022-02-01 [1] CRAN (R 3.6.3)
##    tidyselect             1.1.1      2021-04-30 [1] CRAN (R 3.6.3)
##    tidyverse            * 1.3.1      2021-04-15 [1] CRAN (R 3.6.3)
##    tzdb                   0.2.0      2021-10-27 [1] CRAN (R 3.6.3)
##    UpSetR                 1.4.0      2019-05-22 [1] CRAN (R 3.6.3)
##    usethis                2.1.6      2022-05-25 [2] CRAN (R 3.6.3)
##    utf8                   1.2.2      2021-07-24 [1] CRAN (R 3.6.3)
##    vctrs                  0.3.8      2021-04-29 [1] CRAN (R 3.6.3)
##    vegan                * 2.5-7      2020-11-28 [1] CRAN (R 3.6.3)
##    VennDiagram            1.7.1      2021-12-02 [1] CRAN (R 3.6.3)
##    viridisLite            0.4.0      2021-04-13 [2] CRAN (R 3.6.3)
##    vroom                  1.5.7      2021-11-30 [1] CRAN (R 3.6.3)
##    wesanderson          * 0.3.6.9000 2021-07-21 [1] Github (karthik/wesanderson@651c944)
##    withr                  2.4.3      2021-11-30 [1] CRAN (R 3.6.3)
##    Wrench                 1.4.0      2019-10-29 [1] Bioconductor
##    xfun                   0.23       2021-05-15 [1] CRAN (R 3.6.3)
##    xlsx                 * 0.6.5      2020-11-10 [1] CRAN (R 3.6.3)
##    xlsxjars               0.6.1      2014-08-22 [1] CRAN (R 3.6.3)
##    XMAS                 * 0.0.0.9000 2022-03-23 [1] local
##    XMAS2                  2.1.8.3    2022-11-08 [2] local
##    XML                    3.99-0.3   2020-01-20 [1] CRAN (R 3.6.3)
##    xml2                   1.3.3      2021-11-30 [2] CRAN (R 3.6.3)
##    xtable                 1.8-4      2019-04-21 [1] CRAN (R 3.6.3)
##    XVector                0.26.0     2019-10-29 [2] Bioconductor
##    xviz                 * 1.1.0      2021-01-14 [1] local
##    yaml                   2.2.2      2022-01-25 [1] CRAN (R 3.6.3)
##    zlibbioc               1.32.0     2019-10-29 [2] Bioconductor
##    zoo                    1.8-9      2021-03-09 [1] CRAN (R 3.6.3)
## 
##  [1] /share/home/tongbangzhuo/R/x86_64-pc-linux-gnu-library/3.6
##  [2] /opt/R-3.6.3/lib/R/library
## 
##  R â”€â”€ Package was removed from disk.
## 
## â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="functional-analysis-of-kegg.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="msea.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/FunctionAnalysis/blob/main/03-Functional_Analysis_Target_database_analysis.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Function standard analysis workflow.pdf", "Function standard analysis workflow.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection",
"scroll_highlight": true,
"toc_depth": 3
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
