<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Functional Analysis of Metacyc database | Functional analysis</title>
  <meta name="description" content="Tutorial for MGS functional analysis" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Functional Analysis of Metacyc database | Functional analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Tutorial for MGS functional analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Functional Analysis of Metacyc database | Functional analysis" />
  
  <meta name="twitter:description" content="Tutorial for MGS functional analysis" />
  

<meta name="author" content="Bangzhuo Tong" />


<meta name="date" content="2022-06-29" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="preprocess-of-functional-analysis-of-kegg.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="assets/style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Function standard analysis workflow</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html"><i class="fa fa-check"></i><b>2</b> Functional Analysis of Metacyc database</a><ul>
<li class="chapter" data-level="2.1" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#environment-setup"><i class="fa fa-check"></i><b>2.1</b> Environment setup</a></li>
<li class="chapter" data-level="2.2" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#load-data"><i class="fa fa-check"></i><b>2.2</b> load data</a></li>
<li class="chapter" data-level="2.3" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#remove-unppaed-pathways"><i class="fa fa-check"></i><b>2.3</b> Remove unppaed pathways</a></li>
<li class="chapter" data-level="2.4" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#plot-beta-diversity"><i class="fa fa-check"></i><b>2.4</b> Plot Beta diversity</a></li>
<li class="chapter" data-level="2.5" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#permanova-test"><i class="fa fa-check"></i><b>2.5</b> Permanova test</a></li>
<li class="chapter" data-level="2.6" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#differential-analysisda"><i class="fa fa-check"></i><b>2.6</b> Differential analysis(DA)</a><ul>
<li class="chapter" data-level="2.6.1" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#preprocess-before-da"><i class="fa fa-check"></i><b>2.6.1</b> Preprocess before DA</a></li>
<li class="chapter" data-level="2.6.2" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#lefse"><i class="fa fa-check"></i><b>2.6.2</b> LefSE</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#bacteria-contributing-to-differential-pathways"><i class="fa fa-check"></i><b>2.7</b> Bacteria contributing to differential pathways</a></li>
<li class="chapter" data-level="2.8" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#plot-bacteria-contribution"><i class="fa fa-check"></i><b>2.8</b> Plot Bacteria Contribution</a></li>
<li class="chapter" data-level="2.9" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#gene-and-ensyme-involed-in-differential-pathways"><i class="fa fa-check"></i><b>2.9</b> Gene and Ensyme involed in differential pathways</a><ul>
<li class="chapter" data-level="2.9.1" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#load-tables-to-mapping-pathway-to-gene-and-enzyme"><i class="fa fa-check"></i><b>2.9.1</b> load tables to mapping pathway to gene and enzyme</a></li>
<li class="chapter" data-level="2.9.2" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#mapping-pathway-to-related-gene-and-reaction"><i class="fa fa-check"></i><b>2.9.2</b> Mapping pathway to related gene and reaction</a></li>
</ul></li>
<li class="chapter" data-level="2.10" data-path="functional-analysis-of-metacyc-database.html"><a href="functional-analysis-of-metacyc-database.html#session-info"><i class="fa fa-check"></i><b>2.10</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="preprocess-of-functional-analysis-of-kegg.html"><a href="preprocess-of-functional-analysis-of-kegg.html"><i class="fa fa-check"></i><b>3</b> Preprocess of Functional Analysis of KEGG</a><ul>
<li class="chapter" data-level="3.1" data-path="preprocess-of-functional-analysis-of-kegg.html"><a href="preprocess-of-functional-analysis-of-kegg.html#environment-setup-1"><i class="fa fa-check"></i><b>3.1</b> Environment Setup</a></li>
<li class="chapter" data-level="3.2" data-path="preprocess-of-functional-analysis-of-kegg.html"><a href="preprocess-of-functional-analysis-of-kegg.html#process-pipeline-output"><i class="fa fa-check"></i><b>3.2</b> Process pipeline output</a><ul>
<li class="chapter" data-level="3.2.1" data-path="preprocess-of-functional-analysis-of-kegg.html"><a href="preprocess-of-functional-analysis-of-kegg.html#merge-data-from-humann2-pipeline-kegg-database"><i class="fa fa-check"></i><b>3.2.1</b> Merge data from humann2 pipeline (Kegg database)</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="preprocess-of-functional-analysis-of-kegg.html"><a href="preprocess-of-functional-analysis-of-kegg.html#session-info-1"><i class="fa fa-check"></i><b>3.3</b> Session info</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html"><i class="fa fa-check"></i><b>4</b> Functional Analysis of KEGG</a><ul>
<li class="chapter" data-level="4.1" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#environment-setup-2"><i class="fa fa-check"></i><b>4.1</b> Environment setup</a></li>
<li class="chapter" data-level="4.2" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#read-in-processed-profile-table-from-script-functional_analysis_kegg_pathway_preprocess.rmd"><i class="fa fa-check"></i><b>4.2</b> Read in processed profile table from script Functional_Analysis_Kegg_Pathway_preprocess.Rmd</a></li>
<li class="chapter" data-level="4.3" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#pre-process-pathway-data"><i class="fa fa-check"></i><b>4.3</b> Pre-process pathway data</a><ul>
<li class="chapter" data-level="4.3.1" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#remove-the-pathways-whose-abundance-are-0-across-all-samples"><i class="fa fa-check"></i><b>4.3.1</b> Remove the pathways whose abundance are 0 across all samples</a></li>
<li class="chapter" data-level="4.3.2" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#rescale-data"><i class="fa fa-check"></i><b>4.3.2</b> Rescale data</a></li>
<li class="chapter" data-level="4.3.3" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#remove-unammped-entry"><i class="fa fa-check"></i><b>4.3.3</b> Remove UNAMMPED entry</a></li>
<li class="chapter" data-level="4.3.4" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#filter-low-abundance-pathway"><i class="fa fa-check"></i><b>4.3.4</b> Filter low abundance pathway</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#standard-analysis"><i class="fa fa-check"></i><b>4.4</b> Standard Analysis</a></li>
<li class="chapter" data-level="4.5" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#differential-analysis"><i class="fa fa-check"></i><b>4.5</b> Differential analysis</a><ul>
<li class="chapter" data-level="4.5.1" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#filter-low-prevalence-pathway"><i class="fa fa-check"></i><b>4.5.1</b> Filter low prevalence pathway</a></li>
<li class="chapter" data-level="4.5.2" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#da"><i class="fa fa-check"></i><b>4.5.2</b> DA</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="functional-analysis-of-kegg.html"><a href="functional-analysis-of-kegg.html#session-information"><i class="fa fa-check"></i><b>4.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="preprocess-of-functional-analysis-of-targetdatabase.html"><a href="preprocess-of-functional-analysis-of-targetdatabase.html"><i class="fa fa-check"></i><b>5</b> Preprocess of Functional Analysis of TargetDatabase</a><ul>
<li class="chapter" data-level="5.1" data-path="preprocess-of-functional-analysis-of-targetdatabase.html"><a href="preprocess-of-functional-analysis-of-targetdatabase.html#preprocess---merge-sample-data"><i class="fa fa-check"></i><b>5.1</b> Preprocess - Merge sample data</a><ul>
<li class="chapter" data-level="5.1.1" data-path="preprocess-of-functional-analysis-of-targetdatabase.html"><a href="preprocess-of-functional-analysis-of-targetdatabase.html#use-script-merge_metaphlan_tables.py-from-humann2-to-merge-all-gene_family-profile-files.-1"><i class="fa fa-check"></i><b>5.1.1</b> Use Script merge_metaphlan_tables.py from humann2 to merge all gene_family profile files.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html"><i class="fa fa-check"></i><b>6</b> Functional Analysis of TargetDatabase</a><ul>
<li class="chapter" data-level="6.1" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#environment-setup-3"><i class="fa fa-check"></i><b>6.1</b> Environment setup</a></li>
<li class="chapter" data-level="6.2" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#use-card-data-as-example"><i class="fa fa-check"></i><b>6.2</b> Use card data as example</a><ul>
<li class="chapter" data-level="6.2.1" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#read-in-card-data"><i class="fa fa-check"></i><b>6.2.1</b> Read in card data</a></li>
<li class="chapter" data-level="6.2.2" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#split-strain-info"><i class="fa fa-check"></i><b>6.2.2</b> Split strain info</a></li>
<li class="chapter" data-level="6.2.3" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#rescale-card-data"><i class="fa fa-check"></i><b>6.2.3</b> Rescale card data</a></li>
<li class="chapter" data-level="6.2.4" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#remove-unmapped-row"><i class="fa fa-check"></i><b>6.2.4</b> Remove UNMAPPED row</a></li>
<li class="chapter" data-level="6.2.5" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#filter-low-abundance-data"><i class="fa fa-check"></i><b>6.2.5</b> Filter low abundance data</a></li>
<li class="chapter" data-level="6.2.6" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#basic-analysis"><i class="fa fa-check"></i><b>6.2.6</b> Basic analysis</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#differential-analysis-1"><i class="fa fa-check"></i><b>6.3</b> Differential analysis</a><ul>
<li class="chapter" data-level="6.3.1" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#filter-low-prevalence-pathway-1"><i class="fa fa-check"></i><b>6.3.1</b> Filter low prevalence pathway</a></li>
<li class="chapter" data-level="6.3.2" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#da-1"><i class="fa fa-check"></i><b>6.3.2</b> DA</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="functional-analysis-of-targetdatabase.html"><a href="functional-analysis-of-targetdatabase.html#session-info-2"><i class="fa fa-check"></i><b>6.4</b> Session info</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Functional analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="functional-analysis-of-metacyc-database" class="section level1">
<h1><span class="header-section-number">Chapter 2</span> Functional Analysis of Metacyc database</h1>
<p>This demo provides guidance to standard data analysis of pathway data. Humman2 is used to identify pathways with MetaCyc database.<br />
There are five 7 steps in standard MGS data anlaysis process:</p>
<ul>
<li>load Humman2 output to Rstudio for analysis<br />
</li>
<li>remove unintegrated and unmapped pathways from dataset<br />
</li>
<li>beta diversity<br />
</li>
<li>Permanova test<br />
</li>
<li>Prevalance filtering<br />
</li>
<li>differential analysis<br />
</li>
<li>bacteria contribution to pathway<br />
</li>
<li>reaction(enzyme) and gene to pathway</li>
</ul>
<div id="environment-setup" class="section level2">
<h2><span class="header-section-number">2.1</span> Environment setup</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="functional-analysis-of-metacyc-database.html#cb1-1"></a>knitr<span class="op">::</span>opts_chunk<span class="op">$</span><span class="kw">set</span>(<span class="dt">warning =</span> <span class="ot">FALSE</span>, <span class="dt">message =</span> <span class="ot">FALSE</span>,<span class="dt">out.width=</span><span class="st">&quot;100%&quot;</span>)</span>
<span id="cb1-2"><a href="functional-analysis-of-metacyc-database.html#cb1-2"></a></span>
<span id="cb1-3"><a href="functional-analysis-of-metacyc-database.html#cb1-3"></a><span class="kw">library</span>(XMAS)</span>
<span id="cb1-4"><a href="functional-analysis-of-metacyc-database.html#cb1-4"></a><span class="kw">library</span>(xviz)</span>
<span id="cb1-5"><a href="functional-analysis-of-metacyc-database.html#cb1-5"></a><span class="kw">library</span>(data.table)</span>
<span id="cb1-6"><a href="functional-analysis-of-metacyc-database.html#cb1-6"></a><span class="kw">library</span>(plotly)</span>
<span id="cb1-7"><a href="functional-analysis-of-metacyc-database.html#cb1-7"></a><span class="kw">library</span>(xlsx)</span>
<span id="cb1-8"><a href="functional-analysis-of-metacyc-database.html#cb1-8"></a><span class="kw">library</span>(magrittr)</span>
<span id="cb1-9"><a href="functional-analysis-of-metacyc-database.html#cb1-9"></a></span>
<span id="cb1-10"><a href="functional-analysis-of-metacyc-database.html#cb1-10"></a><span class="kw">source</span>(<span class="st">&quot;/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/RScript/permanova.R&quot;</span>)</span>
<span id="cb1-11"><a href="functional-analysis-of-metacyc-database.html#cb1-11"></a><span class="kw">source</span>(<span class="st">&quot;/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/RScript/pathway2gene.R&quot;</span>)</span></code></pre></div>
</div>
<div id="load-data" class="section level2">
<h2><span class="header-section-number">2.2</span> load data</h2>
<ul>
<li><strong>metadata</strong></li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="functional-analysis-of-metacyc-database.html#cb2-1"></a>metadata &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/Funtional/metadata.txt&quot;</span>,<span class="dt">header =</span> <span class="ot">TRUE</span>,<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>) <span class="op">%&gt;%</span></span>
<span id="cb2-2"><a href="functional-analysis-of-metacyc-database.html#cb2-2"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">seqid2 =</span> seqid) <span class="op">%&gt;%</span><span class="st">  </span><span class="co"># keep seqid in the metadata</span></span>
<span id="cb2-3"><a href="functional-analysis-of-metacyc-database.html#cb2-3"></a><span class="st">  </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;seqid2&quot;</span>) </span>
<span id="cb2-4"><a href="functional-analysis-of-metacyc-database.html#cb2-4"></a></span>
<span id="cb2-5"><a href="functional-analysis-of-metacyc-database.html#cb2-5"></a><span class="kw">rownames</span>(metadata)</span></code></pre></div>
<pre><code>##  [1] &quot;4371&quot; &quot;4368&quot; &quot;4416&quot; &quot;4367&quot; &quot;4373&quot; &quot;4372&quot; &quot;4389&quot; &quot;4410&quot; &quot;4412&quot; &quot;4413&quot; &quot;4386&quot; &quot;4391&quot; &quot;4390&quot;</code></pre>
<ul>
<li><strong>Humann2</strong>: the unstratified pathway table generated by hummann2 is used for analysis first.</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="functional-analysis-of-metacyc-database.html#cb4-1"></a>pathwaydata &lt;-<span class="st"> </span><span class="kw">read_delim</span>(<span class="st">&quot;/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/Funtional/merged_relab_pathabundance_unstratified.tsv&quot;</span>, <span class="dt">delim =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb4-2"><a href="functional-analysis-of-metacyc-database.html#cb4-2"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="st">`</span><span class="dt"># Pathway</span><span class="st">`</span> =<span class="st"> </span><span class="kw">str_split_fixed</span>(<span class="st">`</span><span class="dt"># Pathway</span><span class="st">`</span>, <span class="st">&quot;:&quot;</span>, <span class="dv">2</span>)[,<span class="dv">1</span>]) <span class="op">%&gt;%</span><span class="st">           </span><span class="co"># do not edit </span></span>
<span id="cb4-3"><a href="functional-analysis-of-metacyc-database.html#cb4-3"></a><span class="st">  </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;# Pathway&quot;</span>) <span class="op">%&gt;%</span><span class="st">                                          </span><span class="co"># do not edit</span></span>
<span id="cb4-4"><a href="functional-analysis-of-metacyc-database.html#cb4-4"></a><span class="st">  </span><span class="kw">t</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st">                                                  </span><span class="co"># do not edit</span></span>
<span id="cb4-5"><a href="functional-analysis-of-metacyc-database.html#cb4-5"></a><span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="st">&quot;seq_id&quot;</span>) <span class="op">%&gt;%</span><span class="st">                                             </span><span class="co"># do not edit</span></span>
<span id="cb4-6"><a href="functional-analysis-of-metacyc-database.html#cb4-6"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">seq_id=</span><span class="kw">str_extract</span>(seq_id, <span class="st">&quot;</span><span class="ch">\\</span><span class="st">d{4}&quot;</span>)) <span class="op">%&gt;%</span><span class="st">                             </span><span class="co"># do not edit </span></span>
<span id="cb4-7"><a href="functional-analysis-of-metacyc-database.html#cb4-7"></a><span class="st">  </span><span class="kw">filter</span>(seq_id <span class="op">%in%</span><span class="st"> </span>metadata<span class="op">$</span>seqid) <span class="op">%&gt;%</span><span class="st">                                       </span><span class="co"># *** only keep samples in metadata, edit if needed ***</span></span>
<span id="cb4-8"><a href="functional-analysis-of-metacyc-database.html#cb4-8"></a><span class="st">  </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;seq_id&quot;</span>)   <span class="op">%&gt;%</span><span class="st">                                           </span><span class="co"># do not edit</span></span>
<span id="cb4-9"><a href="functional-analysis-of-metacyc-database.html#cb4-9"></a><span class="st">  </span><span class="kw">t</span>() <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span><span class="st">                                                  </span><span class="co"># do not edit</span></span>
<span id="cb4-10"><a href="functional-analysis-of-metacyc-database.html#cb4-10"></a><span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="st">&quot;pathway&quot;</span>) <span class="op">%&gt;%</span><span class="st">                                            </span><span class="co"># do not edit</span></span>
<span id="cb4-11"><a href="functional-analysis-of-metacyc-database.html#cb4-11"></a><span class="st">  </span><span class="kw">filter</span>(<span class="kw">rowSums</span>(.[,<span class="op">-</span><span class="dv">1</span>]) <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>) <span class="op">%&gt;%</span><span class="st">                                              </span><span class="co"># do not edit</span></span>
<span id="cb4-12"><a href="functional-analysis-of-metacyc-database.html#cb4-12"></a><span class="st">  </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;pathway&quot;</span>)                                                <span class="co"># do not edit</span></span>
<span id="cb4-13"><a href="functional-analysis-of-metacyc-database.html#cb4-13"></a></span>
<span id="cb4-14"><a href="functional-analysis-of-metacyc-database.html#cb4-14"></a><span class="kw">colnames</span>(pathwaydata)</span></code></pre></div>
<pre><code>##  [1] &quot;4367&quot; &quot;4368&quot; &quot;4371&quot; &quot;4372&quot; &quot;4373&quot; &quot;4386&quot; &quot;4389&quot; &quot;4390&quot; &quot;4391&quot; &quot;4410&quot; &quot;4412&quot; &quot;4413&quot; &quot;4416&quot;</code></pre>
</div>
<div id="remove-unppaed-pathways" class="section level2">
<h2><span class="header-section-number">2.3</span> Remove unppaed pathways</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="functional-analysis-of-metacyc-database.html#cb6-1"></a>pathway.clean =<span class="st"> </span>pathwaydata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb6-2"><a href="functional-analysis-of-metacyc-database.html#cb6-2"></a><span class="st">  </span><span class="kw">rownames_to_column</span>(<span class="st">&quot;pathway&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb6-3"><a href="functional-analysis-of-metacyc-database.html#cb6-3"></a><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span>pathway <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;UNMAPPED&quot;</span>,<span class="st">&quot;UNINTEGRATED&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb6-4"><a href="functional-analysis-of-metacyc-database.html#cb6-4"></a><span class="st">  </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;pathway&quot;</span>)</span></code></pre></div>
</div>
<div id="plot-beta-diversity" class="section level2">
<h2><span class="header-section-number">2.4</span> Plot Beta diversity</h2>
<p>In this demo, we plot PCoA with Bray-Curtis distance as example.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="functional-analysis-of-metacyc-database.html#cb7-1"></a><span class="co"># plot_beta_diversity is a fucntion in XVIZ working on calculating the beta diversity distance and visualzing output  </span></span>
<span id="cb7-2"><a href="functional-analysis-of-metacyc-database.html#cb7-2"></a><span class="co"># input of plot_beta_diversity is a phyloseq object</span></span>
<span id="cb7-3"><a href="functional-analysis-of-metacyc-database.html#cb7-3"></a></span>
<span id="cb7-4"><a href="functional-analysis-of-metacyc-database.html#cb7-4"></a><span class="kw">phyloseq</span>(<span class="kw">otu_table</span>(pathway.clean, <span class="dt">taxa_are_rows =</span> T), <span class="kw">sample_data</span>(metadata)) <span class="op">%&gt;%</span></span>
<span id="cb7-5"><a href="functional-analysis-of-metacyc-database.html#cb7-5"></a><span class="st">  </span><span class="kw">plot_beta_diversity</span>(<span class="dt">feature =</span> <span class="st">&quot;Group&quot;</span>,<span class="dt">add_pc1 =</span> <span class="ot">TRUE</span>, <span class="dt">add_pc2 =</span> <span class="ot">TRUE</span>)</span></code></pre></div>
<p><img src="Function-standard-analysis-workflow_files/figure-html/beta_diversity-1.png" width="100%" /></p>
</div>
<div id="permanova-test" class="section level2">
<h2><span class="header-section-number">2.5</span> Permanova test</h2>
<p>The first output is dispersion test and the second output is PERMANOVA test.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="functional-analysis-of-metacyc-database.html#cb8-1"></a><span class="co"># run_permanova_betadisp is a function in XMAS working on doing dispersion test and PERMANOVA test.  </span></span>
<span id="cb8-2"><a href="functional-analysis-of-metacyc-database.html#cb8-2"></a></span>
<span id="cb8-3"><a href="functional-analysis-of-metacyc-database.html#cb8-3"></a><span class="kw">phyloseq</span>(<span class="kw">otu_table</span>(pathway.clean, <span class="dt">taxa_are_rows =</span> T), <span class="kw">sample_data</span>(metadata)) <span class="op">%&gt;%</span></span>
<span id="cb8-4"><a href="functional-analysis-of-metacyc-database.html#cb8-4"></a><span class="st">  </span><span class="kw">run_permanova_betadisp</span>(<span class="dt">vars =</span> <span class="st">&quot;Group&quot;</span>)</span></code></pre></div>
<pre><code>## $betadisp_res
##   variable p_value                       analysis
## 1    Group   0.101 beta_dispersion_permutation999
## 
## $permanova_res
##   variable p_value        R2                 analysis
## 1    Group   0.157 0.1082203 permanova_permutation999</code></pre>
</div>
<div id="differential-analysisda" class="section level2">
<h2><span class="header-section-number">2.6</span> Differential analysis(DA)</h2>
<div id="preprocess-before-da" class="section level3">
<h3><span class="header-section-number">2.6.1</span> Preprocess before DA</h3>
<p>We would remove pathways apperaing in less than max(2 , 5% of samples) from data set before doing analysis.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="functional-analysis-of-metacyc-database.html#cb10-1"></a>pathway.filter &lt;-<span class="st"> </span>XMAS<span class="op">::</span><span class="kw">filter_prevalence</span>(<span class="dt">otu_table =</span> pathway.clean,</span>
<span id="cb10-2"><a href="functional-analysis-of-metacyc-database.html#cb10-2"></a>                                    <span class="dt">metadata =</span> metadata.clean,</span>
<span id="cb10-3"><a href="functional-analysis-of-metacyc-database.html#cb10-3"></a>                                    <span class="dt">threshold =</span> <span class="fl">0.05</span>,</span>
<span id="cb10-4"><a href="functional-analysis-of-metacyc-database.html#cb10-4"></a>                                    <span class="dt">taxa_are_rows =</span> <span class="ot">TRUE</span>)</span>
<span id="cb10-5"><a href="functional-analysis-of-metacyc-database.html#cb10-5"></a></span>
<span id="cb10-6"><a href="functional-analysis-of-metacyc-database.html#cb10-6"></a><span class="kw">nrow</span>(pathway.clean)</span></code></pre></div>
<pre><code>## [1] 429</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="functional-analysis-of-metacyc-database.html#cb12-1"></a><span class="kw">nrow</span>(pathway.filter)</span></code></pre></div>
<pre><code>## [1] 367</code></pre>
</div>
<div id="lefse" class="section level3">
<h3><span class="header-section-number">2.6.2</span> LefSE</h3>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="functional-analysis-of-metacyc-database.html#cb14-1"></a>da_pathway =<span class="st">  </span><span class="kw">phyloseq</span>(<span class="kw">otu_table</span>(pathway.filter, <span class="dt">taxa_are_rows =</span> T), <span class="kw">sample_data</span>(metadata)) <span class="op">%&gt;%</span></span>
<span id="cb14-2"><a href="functional-analysis-of-metacyc-database.html#cb14-2"></a><span class="st">  </span><span class="kw">lefse_mgs</span>(<span class="dt">compvar =</span> <span class="st">&quot;Group&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb14-3"><a href="functional-analysis-of-metacyc-database.html#cb14-3"></a><span class="st">  </span>.<span class="op">$</span>DA_features</span>
<span id="cb14-4"><a href="functional-analysis-of-metacyc-database.html#cb14-4"></a></span>
<span id="cb14-5"><a href="functional-analysis-of-metacyc-database.html#cb14-5"></a></span>
<span id="cb14-6"><a href="functional-analysis-of-metacyc-database.html#cb14-6"></a>da_pathway<span class="op">$</span>id <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">str_replace_all</span>(<span class="st">&quot;_&quot;</span>,<span class="st">&quot;-&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="bacteria-contributing-to-differential-pathways" class="section level2">
<h2><span class="header-section-number">2.7</span> Bacteria contributing to differential pathways</h2>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="functional-analysis-of-metacyc-database.html#cb15-1"></a><span class="kw">options</span>(<span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</span>
<span id="cb15-2"><a href="functional-analysis-of-metacyc-database.html#cb15-2"></a>pathway_stratified &lt;-<span class="st"> </span><span class="kw">read_delim</span>(<span class="st">&quot;/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/merged_relab_pathabundance_stratified.pcl&quot;</span>, <span class="dt">delim =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb15-3"><a href="functional-analysis-of-metacyc-database.html#cb15-3"></a><span class="st">    </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;pathwayID&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb15-4"><a href="functional-analysis-of-metacyc-database.html#cb15-4"></a><span class="st">    </span><span class="kw">t</span>() <span class="op">%&gt;%</span></span>
<span id="cb15-5"><a href="functional-analysis-of-metacyc-database.html#cb15-5"></a><span class="st">    </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb15-6"><a href="functional-analysis-of-metacyc-database.html#cb15-6"></a><span class="st">    </span><span class="kw">rownames_to_column</span>(<span class="st">&quot;seq_id&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb15-7"><a href="functional-analysis-of-metacyc-database.html#cb15-7"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">seq_id=</span><span class="kw">str_extract</span>(seq_id, <span class="st">&quot;</span><span class="ch">\\</span><span class="st">d{4}&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb15-8"><a href="functional-analysis-of-metacyc-database.html#cb15-8"></a><span class="st">    </span><span class="kw">filter</span>(seq_id <span class="op">%in%</span><span class="st"> </span>metadata<span class="op">$</span>seqid) <span class="op">%&gt;%</span></span>
<span id="cb15-9"><a href="functional-analysis-of-metacyc-database.html#cb15-9"></a><span class="st">    </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;seq_id&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb15-10"><a href="functional-analysis-of-metacyc-database.html#cb15-10"></a><span class="st">    </span><span class="kw">t</span>() <span class="op">%&gt;%</span></span>
<span id="cb15-11"><a href="functional-analysis-of-metacyc-database.html#cb15-11"></a><span class="st">    </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb15-12"><a href="functional-analysis-of-metacyc-database.html#cb15-12"></a><span class="st">    </span>.[<span class="kw">apply</span>(.<span class="op">&gt;</span><span class="dv">0</span>, <span class="dv">1</span>, any),] <span class="op">%&gt;%</span></span>
<span id="cb15-13"><a href="functional-analysis-of-metacyc-database.html#cb15-13"></a><span class="st">  </span><span class="co">#  .[apply(.&gt;0,1, function(x) sum(x) &gt; 4),] %&gt;%</span></span>
<span id="cb15-14"><a href="functional-analysis-of-metacyc-database.html#cb15-14"></a><span class="st">    </span><span class="kw">rownames_to_column</span>(<span class="st">&quot;pathwayID&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb15-15"><a href="functional-analysis-of-metacyc-database.html#cb15-15"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">PathwayID=</span><span class="kw">str_split_fixed</span>(pathwayID,<span class="st">&quot;:&quot;</span>,<span class="dv">2</span>)[,<span class="dv">1</span>]) <span class="op">%&gt;%</span></span>
<span id="cb15-16"><a href="functional-analysis-of-metacyc-database.html#cb15-16"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">Introduce=</span><span class="kw">str_split_fixed</span>(<span class="kw">str_split_fixed</span>(pathwayID,<span class="st">&quot;:&quot;</span>,<span class="dv">2</span>)[,<span class="dv">2</span>],<span class="st">&quot;</span><span class="ch">\\</span><span class="st">|&quot;</span>,<span class="dv">2</span>)[,<span class="dv">1</span>]) <span class="op">%&gt;%</span></span>
<span id="cb15-17"><a href="functional-analysis-of-metacyc-database.html#cb15-17"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">Genus=</span><span class="kw">str_split_fixed</span>(<span class="kw">str_split_fixed</span>(pathwayID,<span class="st">&quot;</span><span class="ch">\\</span><span class="st">|&quot;</span>,<span class="dv">2</span>)[,<span class="dv">2</span>], <span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="dv">2</span>)[,<span class="dv">1</span>]) <span class="op">%&gt;%</span></span>
<span id="cb15-18"><a href="functional-analysis-of-metacyc-database.html#cb15-18"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">Species=</span><span class="kw">str_split_fixed</span>(<span class="kw">str_split_fixed</span>(pathwayID,<span class="st">&quot;</span><span class="ch">\\</span><span class="st">|&quot;</span>,<span class="dv">2</span>)[,<span class="dv">2</span>], <span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>, <span class="dv">2</span>)[,<span class="dv">2</span>]) <span class="op">%&gt;%</span></span>
<span id="cb15-19"><a href="functional-analysis-of-metacyc-database.html#cb15-19"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">Species=</span><span class="kw">str_replace</span>(Species, <span class="st">&quot;s__&quot;</span>, <span class="st">&quot;&quot;</span>))</span>
<span id="cb15-20"><a href="functional-analysis-of-metacyc-database.html#cb15-20"></a></span>
<span id="cb15-21"><a href="functional-analysis-of-metacyc-database.html#cb15-21"></a>pathway_list &lt;-<span class="st"> </span>pathway_stratified <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb15-22"><a href="functional-analysis-of-metacyc-database.html#cb15-22"></a><span class="st">    </span>dplyr<span class="op">::</span><span class="kw">select</span>(PathwayID, Species) <span class="op">%&gt;%</span></span>
<span id="cb15-23"><a href="functional-analysis-of-metacyc-database.html#cb15-23"></a><span class="st">    </span><span class="kw">filter</span>((<span class="op">!</span><span class="kw">str_detect</span>(PathwayID, <span class="st">&quot;UNINTEGRATED&quot;</span>))  <span class="op">&amp;</span><span class="st"> </span>(PathwayID <span class="op">%in%</span><span class="st"> </span>da_pathway<span class="op">$</span>id) <span class="op">&amp;</span><span class="st"> </span>Species<span class="op">!=</span><span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb15-24"><a href="functional-analysis-of-metacyc-database.html#cb15-24"></a><span class="st">    </span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> <span class="st">&quot;Species&quot;</span>,<span class="dt">values_from =</span> <span class="st">&quot;Species&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb15-25"><a href="functional-analysis-of-metacyc-database.html#cb15-25"></a><span class="st">    </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;PathwayID&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>.[<span class="kw">apply</span>(.,<span class="dv">1</span>,<span class="cf">function</span>(x) <span class="kw">sum</span>(<span class="op">!</span><span class="kw">is.na</span>(x))) <span class="op">&gt;=</span><span class="st"> </span><span class="dv">2</span>,] <span class="op">%&gt;%</span></span>
<span id="cb15-26"><a href="functional-analysis-of-metacyc-database.html#cb15-26"></a><span class="st">    </span><span class="kw">rownames</span>()</span>
<span id="cb15-27"><a href="functional-analysis-of-metacyc-database.html#cb15-27"></a>    </span>
<span id="cb15-28"><a href="functional-analysis-of-metacyc-database.html#cb15-28"></a></span>
<span id="cb15-29"><a href="functional-analysis-of-metacyc-database.html#cb15-29"></a><span class="kw">read_delim</span>(<span class="st">&quot;/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/Funtional/merged_relab_pathabundance_stratified.tsv&quot;</span>, <span class="dt">delim =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb15-30"><a href="functional-analysis-of-metacyc-database.html#cb15-30"></a><span class="st">    </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;# Pathway&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb15-31"><a href="functional-analysis-of-metacyc-database.html#cb15-31"></a><span class="st">    </span><span class="kw">t</span>() <span class="op">%&gt;%</span></span>
<span id="cb15-32"><a href="functional-analysis-of-metacyc-database.html#cb15-32"></a><span class="st">    </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb15-33"><a href="functional-analysis-of-metacyc-database.html#cb15-33"></a><span class="st">    </span><span class="kw">rownames_to_column</span>(<span class="st">&quot;seqid&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb15-34"><a href="functional-analysis-of-metacyc-database.html#cb15-34"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">seqid=</span><span class="kw">str_extract</span>(seqid, <span class="st">&quot;</span><span class="ch">\\</span><span class="st">d{4}&quot;</span>)) <span class="op">%&gt;%</span></span>
<span id="cb15-35"><a href="functional-analysis-of-metacyc-database.html#cb15-35"></a><span class="st">    </span><span class="kw">merge</span>(metadata,.,  <span class="dt">by=</span><span class="st">&quot;seqid&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb15-36"><a href="functional-analysis-of-metacyc-database.html#cb15-36"></a><span class="st">    </span><span class="kw">filter</span>(seqid <span class="op">%in%</span><span class="st"> </span>metadata<span class="op">$</span>seqid) <span class="op">%&gt;%</span></span>
<span id="cb15-37"><a href="functional-analysis-of-metacyc-database.html#cb15-37"></a><span class="st">    </span><span class="kw">column_to_rownames</span>(<span class="st">&quot;seqid&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb15-38"><a href="functional-analysis-of-metacyc-database.html#cb15-38"></a><span class="st">    </span><span class="kw">t</span>() <span class="op">%&gt;%</span></span>
<span id="cb15-39"><a href="functional-analysis-of-metacyc-database.html#cb15-39"></a><span class="st">    </span><span class="kw">as.data.frame</span>() <span class="op">%&gt;%</span></span>
<span id="cb15-40"><a href="functional-analysis-of-metacyc-database.html#cb15-40"></a><span class="st">    </span><span class="kw">rownames_to_column</span>(<span class="st">&quot;pathwayID&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb15-41"><a href="functional-analysis-of-metacyc-database.html#cb15-41"></a><span class="st">    </span><span class="kw">write.table</span>(.,<span class="st">&quot;/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/merged_relab_pathabundance_stratified.pcl&quot;</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>, <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</span>
<span id="cb15-42"><a href="functional-analysis-of-metacyc-database.html#cb15-42"></a></span>
<span id="cb15-43"><a href="functional-analysis-of-metacyc-database.html#cb15-43"></a><span class="cf">for</span> (i <span class="cf">in</span> pathway_list) {</span>
<span id="cb15-44"><a href="functional-analysis-of-metacyc-database.html#cb15-44"></a>    commond_text &lt;-<span class="st"> &quot;python3.5 ~/tongbangzhuo_Xbiome1/Script/humann2_barplot.py --input /home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/Funtional/merged_relab_pathabundance_stratified.pcl --focal-feature ${pathwayID} --focal-metadatum Group --last-metadatum Group -d 8 4 --output /share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/${pathwayID}.png -e 0.8&quot;</span></span>
<span id="cb15-45"><a href="functional-analysis-of-metacyc-database.html#cb15-45"></a>    commond_text &lt;-<span class="st"> </span>stringr<span class="op">::</span><span class="kw">str_interp</span>(commond_text, <span class="kw">list</span>(<span class="dt">pathwayID =</span> i))</span>
<span id="cb15-46"><a href="functional-analysis-of-metacyc-database.html#cb15-46"></a>    <span class="kw">system</span>(commond_text)</span>
<span id="cb15-47"><a href="functional-analysis-of-metacyc-database.html#cb15-47"></a>}</span></code></pre></div>
</div>
<div id="plot-bacteria-contribution" class="section level2">
<h2><span class="header-section-number">2.8</span> Plot Bacteria Contribution</h2>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="functional-analysis-of-metacyc-database.html#cb16-1"></a>image_path &lt;-<span class="st"> </span><span class="ot">NULL</span></span>
<span id="cb16-2"><a href="functional-analysis-of-metacyc-database.html#cb16-2"></a><span class="co"># for (image in list.files(&quot;/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/&quot;)){</span></span>
<span id="cb16-3"><a href="functional-analysis-of-metacyc-database.html#cb16-3"></a><span class="co">#     if (str_detect(image,&quot;.png$&quot;)){</span></span>
<span id="cb16-4"><a href="functional-analysis-of-metacyc-database.html#cb16-4"></a><span class="co">#         image_path &lt;- c(image_path, paste0(&quot;/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/&quot;,image))</span></span>
<span id="cb16-5"><a href="functional-analysis-of-metacyc-database.html#cb16-5"></a><span class="co">#     }</span></span>
<span id="cb16-6"><a href="functional-analysis-of-metacyc-database.html#cb16-6"></a><span class="co"># }</span></span>
<span id="cb16-7"><a href="functional-analysis-of-metacyc-database.html#cb16-7"></a>image_path &lt;-<span class="st"> </span><span class="kw">system</span>(<span class="st">&#39;ls /share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/*.png&#39;</span>, <span class="dt">intern =</span> <span class="ot">TRUE</span>)</span>
<span id="cb16-8"><a href="functional-analysis-of-metacyc-database.html#cb16-8"></a></span>
<span id="cb16-9"><a href="functional-analysis-of-metacyc-database.html#cb16-9"></a><span class="co">## In the graph(s) below, green KO are those KOs enriched in GroupB, red KO are those KOs enriched in GroupA</span></span>
<span id="cb16-10"><a href="functional-analysis-of-metacyc-database.html#cb16-10"></a>knitr<span class="op">::</span><span class="kw">include_graphics</span>(image_path)</span></code></pre></div>
<p><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/ANAGLYCOLYSIS-PWY.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/ARGININE-SYN4-PWY.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/ARGSYN-PWY.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/CALVIN-PWY.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/COBALSYN-PWY.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/GLUCUROCAT-PWY.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/HISTSYN-PWY.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/P125-PWY.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/PWY0-1586.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/PWY0-162.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/PWY0-166.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/PWY-1269.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/PWY-6527.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/PWY-6545.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/PWY-6606.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/PWY-6703.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/PWY-6897.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/PWY-7184.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/PWY-7400.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/PWY-922.png" width="100%" /><img src="/share/projects/SOP/Functional_Analysis/Tongbangzhuo/Phase1/Metacyc/Demodata/pathway_bacteria/THISYNARA-PWY.png" width="100%" /></p>
</div>
<div id="gene-and-ensyme-involed-in-differential-pathways" class="section level2">
<h2><span class="header-section-number">2.9</span> Gene and Ensyme involed in differential pathways</h2>
<div id="load-tables-to-mapping-pathway-to-gene-and-enzyme" class="section level3">
<h3><span class="header-section-number">2.9.1</span> load tables to mapping pathway to gene and enzyme</h3>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="functional-analysis-of-metacyc-database.html#cb17-1"></a><span class="co"># gene family</span></span>
<span id="cb17-2"><a href="functional-analysis-of-metacyc-database.html#cb17-2"></a>my_genefamily =<span class="st"> </span><span class="kw">read.table</span>(<span class="st">&quot;/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/Funtional/metacyc_data/all.genefamilies.tsv&quot;</span>,<span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)</span>
<span id="cb17-3"><a href="functional-analysis-of-metacyc-database.html#cb17-3"></a>my_genefamily_unstratified =<span class="st"> </span>my_genefamily <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">str_detect</span>(V1,<span class="st">&quot;[|]&quot;</span>))  </span>
<span id="cb17-4"><a href="functional-analysis-of-metacyc-database.html#cb17-4"></a></span>
<span id="cb17-5"><a href="functional-analysis-of-metacyc-database.html#cb17-5"></a>metacyc_pwy_name =<span class="st"> </span><span class="kw">read_delim</span>(<span class="st">&quot;/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/Funtional/metacyc_data/map_metacyc-pwy_name.txt&quot;</span>,<span class="dt">delim =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>,<span class="dt">col_names =</span> <span class="ot">FALSE</span>)</span></code></pre></div>
</div>
<div id="mapping-pathway-to-related-gene-and-reaction" class="section level3">
<h3><span class="header-section-number">2.9.2</span> Mapping pathway to related gene and reaction</h3>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="functional-analysis-of-metacyc-database.html#cb18-1"></a>cur_path =<span class="st"> </span><span class="dv">1</span></span>
<span id="cb18-2"><a href="functional-analysis-of-metacyc-database.html#cb18-2"></a>append_option =<span class="st"> </span><span class="ot">TRUE</span></span>
<span id="cb18-3"><a href="functional-analysis-of-metacyc-database.html#cb18-3"></a>da_pathway_gene_full_list =<span class="st"> </span><span class="kw">list</span>()</span>
<span id="cb18-4"><a href="functional-analysis-of-metacyc-database.html#cb18-4"></a></span>
<span id="cb18-5"><a href="functional-analysis-of-metacyc-database.html#cb18-5"></a></span>
<span id="cb18-6"><a href="functional-analysis-of-metacyc-database.html#cb18-6"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(pathway_list)){</span>
<span id="cb18-7"><a href="functional-analysis-of-metacyc-database.html#cb18-7"></a>  pathway_gene_result =<span class="st"> </span><span class="kw">pathway2gene</span>(pathway_list[i]) </span>
<span id="cb18-8"><a href="functional-analysis-of-metacyc-database.html#cb18-8"></a>  </span>
<span id="cb18-9"><a href="functional-analysis-of-metacyc-database.html#cb18-9"></a>  <span class="cf">if</span>(<span class="kw">is.null</span>(pathway_gene_result)){</span>
<span id="cb18-10"><a href="functional-analysis-of-metacyc-database.html#cb18-10"></a>    cur_path =<span class="st"> </span>cur_path<span class="op">+</span><span class="dv">1</span></span>
<span id="cb18-11"><a href="functional-analysis-of-metacyc-database.html#cb18-11"></a>    <span class="cf">next</span>()</span>
<span id="cb18-12"><a href="functional-analysis-of-metacyc-database.html#cb18-12"></a>  }</span>
<span id="cb18-13"><a href="functional-analysis-of-metacyc-database.html#cb18-13"></a>  </span>
<span id="cb18-14"><a href="functional-analysis-of-metacyc-database.html#cb18-14"></a>  <span class="cf">else</span>{</span>
<span id="cb18-15"><a href="functional-analysis-of-metacyc-database.html#cb18-15"></a>    pathway_gene_result =<span class="st"> </span>pathway_gene_result <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(genefamily <span class="op">%in%</span><span class="st"> </span>my_genefamily<span class="op">$</span>V1)</span>
<span id="cb18-16"><a href="functional-analysis-of-metacyc-database.html#cb18-16"></a>  </span>
<span id="cb18-17"><a href="functional-analysis-of-metacyc-database.html#cb18-17"></a>  <span class="cf">if</span>(cur_path <span class="op">==</span><span class="st"> </span><span class="dv">1</span>){</span>
<span id="cb18-18"><a href="functional-analysis-of-metacyc-database.html#cb18-18"></a>    append_option =<span class="st"> </span><span class="ot">FALSE</span></span>
<span id="cb18-19"><a href="functional-analysis-of-metacyc-database.html#cb18-19"></a>  }</span>
<span id="cb18-20"><a href="functional-analysis-of-metacyc-database.html#cb18-20"></a>  </span>
<span id="cb18-21"><a href="functional-analysis-of-metacyc-database.html#cb18-21"></a>  <span class="cf">if</span>(cur_path <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span>){</span>
<span id="cb18-22"><a href="functional-analysis-of-metacyc-database.html#cb18-22"></a>     append_option =<span class="st"> </span><span class="ot">TRUE</span></span>
<span id="cb18-23"><a href="functional-analysis-of-metacyc-database.html#cb18-23"></a>  }</span>
<span id="cb18-24"><a href="functional-analysis-of-metacyc-database.html#cb18-24"></a>  </span>
<span id="cb18-25"><a href="functional-analysis-of-metacyc-database.html#cb18-25"></a>  <span class="co">#write.xlsx(pathway_gene_result,&quot;/home/xuxiaomin/project/standardized_analytics_workflow_R_function/demo_data/Funtional/pathway_reaction_gene.xlsx&quot;,</span></span>
<span id="cb18-26"><a href="functional-analysis-of-metacyc-database.html#cb18-26"></a>  <span class="co">#           row.names = FALSE, col.names = TRUE,sheetName = pathway_list[i],</span></span>
<span id="cb18-27"><a href="functional-analysis-of-metacyc-database.html#cb18-27"></a>  <span class="co">#           append = append_option)</span></span>
<span id="cb18-28"><a href="functional-analysis-of-metacyc-database.html#cb18-28"></a>  </span>
<span id="cb18-29"><a href="functional-analysis-of-metacyc-database.html#cb18-29"></a>  da_pathway_gene_full_list[[pathway_list[i]]] =<span class="st"> </span>pathway_gene_result</span>
<span id="cb18-30"><a href="functional-analysis-of-metacyc-database.html#cb18-30"></a>    </span>
<span id="cb18-31"><a href="functional-analysis-of-metacyc-database.html#cb18-31"></a>  cur_path =<span class="st"> </span>cur_path <span class="op">+</span><span class="st"> </span><span class="dv">1</span></span>
<span id="cb18-32"><a href="functional-analysis-of-metacyc-database.html#cb18-32"></a>  }  </span>
<span id="cb18-33"><a href="functional-analysis-of-metacyc-database.html#cb18-33"></a>  </span>
<span id="cb18-34"><a href="functional-analysis-of-metacyc-database.html#cb18-34"></a>}</span></code></pre></div>
<p>Show the result of one of the differential pathways for example:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="functional-analysis-of-metacyc-database.html#cb19-1"></a><span class="kw">head</span>(da_pathway_gene_full_list[[<span class="dv">1</span>]])</span></code></pre></div>
<pre><code>##            reaction          genefamily
## 1         RXN-15513 UniRef90_A0A010ZHV5
## 2         RXN-15513     UniRef90_A3N0J2
## 3         RXN-15513     UniRef90_A6L1K6
## 4         RXN-15513     UniRef90_A6L9K8
## 5 GAPOXNPHOSPHN-RXN UniRef90_A0A024L2B0
## 6 GAPOXNPHOSPHN-RXN UniRef90_A0A024L2Z2</code></pre>
</div>
</div>
<div id="session-info" class="section level2">
<h2><span class="header-section-number">2.10</span> Session Info</h2>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="functional-analysis-of-metacyc-database.html#cb21-1"></a>devtools<span class="op">::</span><span class="kw">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 3.6.3 (2020-02-29)
##  os       Ubuntu 16.04.7 LTS
##  system   x86_64, linux-gnu
##  ui       RStudio
##  language (EN)
##  collate  en_IN.UTF-8
##  ctype    en_IN.UTF-8
##  tz       Asia/Hong_Kong
##  date     2022-06-29
##  rstudio  1.1.419 (server)
##  pandoc   2.7.3 @ /usr/bin/ (via rmarkdown)
## 
## ─ Packages ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
##  !  package              * version    date (UTC) lib source
##     abind                  1.4-5      2016-07-21 [1] CRAN (R 3.6.3)
##     ade4                   1.7-17     2021-06-17 [1] CRAN (R 3.6.3)
##     ALDEx2                 1.18.0     2019-10-29 [1] Bioconductor
##     annotate               1.64.0     2019-10-29 [1] Bioconductor
##     AnnotationDbi        * 1.48.0     2019-10-29 [1] Bioconductor
##     ape                    5.5        2021-04-25 [1] CRAN (R 3.6.3)
##     assertthat             0.2.1      2019-03-21 [2] CRAN (R 3.6.3)
##     backports              1.4.1      2021-12-13 [1] CRAN (R 3.6.3)
##     base64enc              0.1-3      2015-07-28 [2] CRAN (R 3.6.3)
##     bayesm                 3.1-4      2019-10-15 [1] CRAN (R 3.6.3)
##     biglm                  0.9-2.1    2020-11-27 [1] CRAN (R 3.6.3)
##     Biobase              * 2.46.0     2019-10-29 [2] Bioconductor
##     BiocGenerics         * 0.32.0     2019-10-29 [2] Bioconductor
##     BiocManager            1.30.16    2021-06-15 [1] CRAN (R 3.6.3)
##     BiocParallel         * 1.20.1     2019-12-21 [2] Bioconductor
##     biomformat             1.14.0     2019-10-29 [1] Bioconductor
##     Biostrings             2.54.0     2019-10-29 [1] Bioconductor
##     bit                    4.0.4      2020-08-04 [1] CRAN (R 3.6.3)
##     bit64                  4.0.5      2020-08-30 [1] CRAN (R 3.6.3)
##     bitops                 1.0-7      2021-04-24 [1] CRAN (R 3.6.3)
##     blob                   1.2.2      2021-07-23 [1] CRAN (R 3.6.3)
##     bookdown               0.24       2021-09-02 [1] CRAN (R 3.6.3)
##     brio                   1.1.3      2021-11-30 [2] CRAN (R 3.6.3)
##     broom                  0.7.12     2022-01-28 [1] CRAN (R 3.6.3)
##     bslib                  0.3.1      2021-10-06 [1] CRAN (R 3.6.3)
##     cachem                 1.0.5      2021-05-15 [1] CRAN (R 3.6.3)
##     callr                  3.7.0      2021-04-20 [2] CRAN (R 3.6.3)
##     car                    3.0-12     2021-11-06 [1] CRAN (R 3.6.3)
##     carData                3.0-4      2020-05-22 [1] CRAN (R 3.6.3)
##     caTools                1.18.2     2021-03-28 [1] CRAN (R 3.6.3)
##     cellranger             1.1.0      2016-07-27 [1] CRAN (R 3.6.3)
##     checkmate              2.0.0      2020-02-06 [1] CRAN (R 3.6.3)
##     circlize             * 0.4.13     2021-06-09 [1] CRAN (R 3.6.3)
##     cli                    3.1.0      2021-10-27 [1] CRAN (R 3.6.3)
##     clue                   0.3-59     2021-04-16 [1] CRAN (R 3.6.3)
##     cluster                2.1.0      2019-06-19 [2] CRAN (R 3.6.3)
##     codetools              0.2-16     2018-12-24 [2] CRAN (R 3.6.3)
##     colorspace             2.0-2      2021-06-24 [1] CRAN (R 3.6.3)
##     ComplexHeatmap       * 2.2.0      2019-10-29 [1] Bioconductor
##     compositions           2.0-2      2021-07-14 [1] CRAN (R 3.6.3)
##     cowplot              * 1.1.1      2020-12-30 [1] CRAN (R 3.6.3)
##     crayon                 1.5.0      2022-02-14 [1] CRAN (R 3.6.3)
##     dada2                * 1.14.1     2020-02-22 [1] Bioconductor
##     data.table           * 1.14.0     2021-02-21 [1] CRAN (R 3.6.3)
##     DBI                    1.1.1      2021-01-15 [1] CRAN (R 3.6.3)
##     dbplyr                 2.1.1      2021-04-06 [1] CRAN (R 3.6.3)
##     DelayedArray         * 0.12.3     2020-04-09 [2] Bioconductor
##     DelayedMatrixStats     1.8.0      2019-10-29 [1] Bioconductor
##     DEoptimR               1.0-9      2021-05-24 [1] CRAN (R 3.6.3)
##     desc                   1.4.1      2022-03-06 [2] CRAN (R 3.6.3)
##     DESeq2               * 1.26.0     2019-10-29 [1] Bioconductor
##     devtools               2.4.3      2021-11-30 [1] CRAN (R 3.6.3)
##     digest                 0.6.29     2021-12-01 [1] CRAN (R 3.6.3)
##     dplyr                * 1.0.6      2021-05-05 [1] CRAN (R 3.6.3)
##     edgeR                  3.28.1     2020-02-26 [1] Bioconductor
##     ellipsis               0.3.2      2021-04-29 [1] CRAN (R 3.6.3)
##     EnhancedVolcano      * 1.4.0      2019-10-29 [1] Bioconductor
##     evaluate               0.15       2022-02-18 [2] CRAN (R 3.6.3)
##     fansi                  1.0.2      2022-01-14 [1] CRAN (R 3.6.3)
##     farver                 2.1.0      2021-02-28 [2] CRAN (R 3.6.3)
##     fastmap                1.1.0      2021-01-25 [1] CRAN (R 3.6.3)
##     fdrtool                1.2.17     2021-11-13 [1] CRAN (R 3.6.3)
##     forcats              * 0.5.1      2021-01-27 [1] CRAN (R 3.6.3)
##     foreach                1.5.2      2022-02-02 [2] CRAN (R 3.6.3)
##     foreign                0.8-75     2020-01-20 [2] CRAN (R 3.6.3)
##     formatR                1.12       2022-03-31 [2] CRAN (R 3.6.3)
##     Formula                1.2-4      2020-10-16 [1] CRAN (R 3.6.3)
##     fs                     1.5.2      2021-12-08 [1] CRAN (R 3.6.3)
##     futile.logger        * 1.4.3      2016-07-10 [2] CRAN (R 3.6.3)
##     futile.options         1.0.1      2018-04-20 [2] CRAN (R 3.6.3)
##     genefilter             1.68.0     2019-10-29 [1] Bioconductor
##     geneplotter            1.64.0     2019-10-29 [1] Bioconductor
##     generics               0.1.2      2022-01-31 [1] CRAN (R 3.6.3)
##     GenomeInfoDb         * 1.22.1     2020-03-27 [2] Bioconductor
##     GenomeInfoDbData       1.2.2      2020-08-24 [2] Bioconductor
##     GenomicAlignments      1.22.1     2019-11-12 [1] Bioconductor
##     GenomicRanges        * 1.38.0     2019-10-29 [2] Bioconductor
##     getopt                 1.20.3     2019-03-22 [1] CRAN (R 3.6.3)
##     GetoptLong             1.0.5      2020-12-15 [1] CRAN (R 3.6.3)
##     ggExtra              * 0.9        2019-08-27 [1] CRAN (R 3.6.3)
##     ggplot2              * 3.3.5      2021-06-25 [1] CRAN (R 3.6.3)
##     ggpubr               * 0.4.0      2020-06-27 [1] CRAN (R 3.6.3)
##     ggrepel              * 0.9.1      2021-01-15 [2] CRAN (R 3.6.3)
##     ggsci                * 2.9        2018-05-14 [1] CRAN (R 3.6.3)
##     ggsignif               0.6.3      2021-09-09 [1] CRAN (R 3.6.3)
##     glmnet                 4.1-2      2021-06-24 [1] CRAN (R 3.6.3)
##     GlobalOptions          0.1.2      2020-06-10 [1] CRAN (R 3.6.3)
##     glue                   1.6.1      2022-01-22 [1] CRAN (R 3.6.3)
##     GMPR                   0.1.3      2021-05-17 [1] local
##     gplots                 3.1.1      2020-11-28 [1] CRAN (R 3.6.3)
##     graph                  1.64.0     2019-10-29 [1] Bioconductor
##     gridExtra              2.3        2017-09-09 [2] CRAN (R 3.6.3)
##     gtable                 0.3.0      2019-03-25 [2] CRAN (R 3.6.3)
##     gtools                 3.9.2      2021-06-06 [1] CRAN (R 3.6.3)
##     haven                  2.4.1      2021-04-23 [1] CRAN (R 3.6.3)
##     highr                  0.9        2021-04-16 [1] CRAN (R 3.6.3)
##     Hmisc                  4.5-0      2021-02-28 [1] CRAN (R 3.6.3)
##     hms                    1.1.1      2021-09-26 [1] CRAN (R 3.6.3)
##     htmlTable              2.3.0      2021-10-12 [1] CRAN (R 3.6.3)
##     htmltools              0.5.2      2021-08-25 [1] CRAN (R 3.6.3)
##     htmlwidgets            1.5.4      2021-09-08 [2] CRAN (R 3.6.3)
##     httpuv                 1.6.1      2021-05-07 [1] CRAN (R 3.6.3)
##     httr                   1.4.3      2022-05-04 [2] CRAN (R 3.6.3)
##     hwriter                1.3.2      2014-09-10 [1] CRAN (R 3.6.3)
##     igraph                 1.3.1      2022-04-20 [2] CRAN (R 3.6.3)
##     IHW                    1.14.0     2019-10-29 [1] Bioconductor
##     IRanges              * 2.20.2     2020-01-13 [2] Bioconductor
##     iterators              1.0.14     2022-02-05 [2] CRAN (R 3.6.3)
##     jpeg                   0.1-9      2021-07-24 [1] CRAN (R 3.6.3)
##     jquerylib              0.1.4      2021-04-26 [1] CRAN (R 3.6.3)
##     jsonlite               1.8.0      2022-02-22 [2] CRAN (R 3.6.3)
##     KEGGgraph              1.46.0     2019-10-29 [1] Bioconductor
##     KEGGREST               1.26.1     2019-11-06 [1] Bioconductor
##     KernSmooth             2.23-16    2019-10-15 [2] CRAN (R 3.6.3)
##     knitr                  1.36       2021-09-29 [1] CRAN (R 3.6.3)
##     labeling               0.4.2      2020-10-20 [2] CRAN (R 3.6.3)
##     lambda.r               1.2.4      2019-09-18 [2] CRAN (R 3.6.3)
##     later                  1.3.0      2021-08-18 [2] CRAN (R 3.6.3)
##     lattice              * 0.20-38    2018-11-04 [2] CRAN (R 3.6.3)
##     latticeExtra           0.6-29     2019-12-19 [1] CRAN (R 3.6.3)
##     lazyeval               0.2.2      2019-03-15 [2] CRAN (R 3.6.3)
##     lifecycle              1.0.1      2021-09-24 [1] CRAN (R 3.6.3)
##     limma                  3.42.2     2020-02-03 [2] Bioconductor
##     locfit                 1.5-9.4    2020-03-25 [1] CRAN (R 3.6.3)
##     lpsymphony             1.14.0     2019-10-29 [1] Bioconductor (R 3.6.3)
##     lubridate              1.7.10     2021-02-26 [1] CRAN (R 3.6.3)
##     Maaslin2               1.7.3      2022-03-23 [1] Github (biobakery/maaslin2@8d090e4)
##     magrittr             * 2.0.2      2022-01-26 [1] CRAN (R 3.6.3)
##     MASS                   7.3-54     2021-05-03 [1] CRAN (R 3.6.3)
##     Matrix                 1.3-4      2021-06-01 [1] CRAN (R 3.6.3)
##     matrixStats          * 0.60.0     2021-07-26 [1] CRAN (R 3.6.3)
##     mbzinb                 0.2        2021-06-23 [1] local
##     memoise                2.0.1      2021-11-26 [2] CRAN (R 3.6.3)
##     metagenomeSeq          1.28.2     2020-02-03 [1] Bioconductor
##     metamicrobiomeR        1.1        2021-02-03 [1] local
##     mgcv                   1.8-31     2019-11-09 [2] CRAN (R 3.6.3)
##     microbiome             1.8.0      2019-10-29 [1] Bioconductor
##     mime                   0.12       2021-09-28 [2] CRAN (R 3.6.3)
##     miniUI                 0.1.1.1    2018-05-18 [1] CRAN (R 3.6.3)
##     modelr                 0.1.8      2020-05-19 [1] CRAN (R 3.6.3)
##     multtest               2.42.0     2019-10-29 [2] Bioconductor
##     munsell                0.5.0      2018-06-12 [2] CRAN (R 3.6.3)
##     mvtnorm                1.1-3      2021-10-08 [1] CRAN (R 3.6.3)
##     nlme                   3.1-144    2020-02-06 [2] CRAN (R 3.6.3)
##     nnet                   7.3-12     2016-02-02 [2] CRAN (R 3.6.3)
##     optparse               1.7.1      2021-10-08 [1] CRAN (R 3.6.3)
##     org.Hs.eg.db         * 3.10.0     2021-12-08 [1] Bioconductor
##     pathview             * 1.26.0     2019-10-29 [1] Bioconductor
##     pcaPP                  1.9-74     2021-04-23 [1] CRAN (R 3.6.3)
##     permute              * 0.9-5      2019-03-12 [1] CRAN (R 3.6.3)
##     phyloseq             * 1.30.0     2019-10-29 [1] Bioconductor
##     pillar                 1.7.0      2022-02-01 [1] CRAN (R 3.6.3)
##     pkgbuild               1.3.1      2021-12-20 [2] CRAN (R 3.6.3)
##     pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 3.6.3)
##     pkgload                1.2.4      2021-11-30 [2] CRAN (R 3.6.3)
##     plotly               * 4.10.0     2021-10-09 [1] CRAN (R 3.6.3)
##     plyr                   1.8.7      2022-03-24 [2] CRAN (R 3.6.3)
##     png                    0.1-7      2013-12-03 [1] CRAN (R 3.6.3)
##     prettyunits            1.1.1      2020-01-24 [2] CRAN (R 3.6.3)
##     processx               3.5.3      2022-03-25 [2] CRAN (R 3.6.3)
##     promises               1.2.0.1    2021-02-11 [2] CRAN (R 3.6.3)
##     protoclust             1.6.3      2019-01-31 [1] CRAN (R 3.6.3)
##     ps                     1.7.0      2022-04-23 [2] CRAN (R 3.6.3)
##     pscl                   1.5.5      2020-03-07 [1] CRAN (R 3.6.3)
##     purrr                * 0.3.4      2020-04-17 [2] CRAN (R 3.6.3)
##     qvalue                 2.18.0     2019-10-29 [1] Bioconductor
##     R6                     2.5.1      2021-08-19 [1] CRAN (R 3.6.3)
##     RAIDA                  1.0        2021-06-23 [1] local
##     ranacapa               0.1.0      2021-06-18 [1] Github (gauravsk/ranacapa@58c0cab)
##     RColorBrewer         * 1.1-3      2022-04-03 [2] CRAN (R 3.6.3)
##     Rcpp                 * 1.0.7      2021-07-07 [1] CRAN (R 3.6.3)
##     RcppParallel           5.1.4      2021-05-04 [1] CRAN (R 3.6.3)
##     RCurl                  1.98-1.6   2022-02-08 [2] CRAN (R 3.6.3)
##     readr                * 2.0.0      2021-07-20 [1] CRAN (R 3.6.3)
##     readxl               * 1.3.1      2019-03-13 [1] CRAN (R 3.6.3)
##     remotes                2.4.2      2021-11-30 [1] CRAN (R 3.6.3)
##     reprex                 2.0.1      2021-08-05 [1] CRAN (R 3.6.3)
##     reshape                0.8.9      2022-04-12 [1] CRAN (R 3.6.3)
##     reshape2             * 1.4.4      2020-04-09 [2] CRAN (R 3.6.3)
##     Rgraphviz              2.30.0     2019-10-29 [1] Bioconductor
##     rhdf5                  2.30.1     2019-11-26 [1] Bioconductor
##     Rhdf5lib               1.8.0      2019-10-29 [1] Bioconductor
##     rJava                  1.0-5      2021-09-24 [1] CRAN (R 3.6.3)
##     rjson                  0.2.20     2018-06-08 [1] CRAN (R 3.6.3)
##  R  rlang                  1.0.2      &lt;NA&gt;       [2] &lt;NA&gt;
##     rmarkdown              2.11       2021-09-14 [1] CRAN (R 3.6.3)
##     robustbase             0.93-9     2021-09-27 [1] CRAN (R 3.6.3)
##     rpart                  4.1-15     2019-04-12 [2] CRAN (R 3.6.3)
##     rprojroot              2.0.2      2020-11-15 [1] CRAN (R 3.6.3)
##     Rsamtools              2.2.3      2020-02-23 [1] Bioconductor
##     RSQLite                2.2.7      2021-04-22 [1] CRAN (R 3.6.3)
##     rstatix                0.7.0      2021-02-13 [1] CRAN (R 3.6.3)
##     rstudioapi             0.13       2020-11-12 [2] CRAN (R 3.6.3)
##     Rtsne                  0.15       2018-11-10 [1] CRAN (R 3.6.3)
##     rvest                  1.0.2      2021-10-16 [1] CRAN (R 3.6.3)
##     S4Vectors            * 0.24.4     2020-04-09 [2] Bioconductor
##     sass                   0.4.0      2021-05-12 [1] CRAN (R 3.6.3)
##     scales                 1.2.0      2022-04-13 [2] CRAN (R 3.6.3)
##     seqinr               * 4.2-8      2021-06-09 [1] CRAN (R 3.6.3)
##     sessioninfo            1.2.2      2021-12-06 [2] CRAN (R 3.6.3)
##     shape                  1.4.6      2021-05-19 [1] CRAN (R 3.6.3)
##     shiny                  1.7.1      2021-10-02 [1] CRAN (R 3.6.3)
##     ShortRead              1.44.3     2020-02-03 [1] Bioconductor
##     slam                   0.1-49     2021-11-17 [1] CRAN (R 3.6.3)
##  VP stringi                1.7.6      2021-08-25 [2] CRAN (R 3.6.3) (on disk 1.7.4)
##     stringr              * 1.4.0      2019-02-10 [2] CRAN (R 3.6.3)
##     SummarizedExperiment * 1.16.1     2019-12-19 [2] Bioconductor
##     survival               3.1-8      2019-12-03 [2] CRAN (R 3.6.3)
##     tensorA                0.36.2     2020-11-19 [1] CRAN (R 3.6.3)
##     testthat               3.1.4      2022-04-26 [2] CRAN (R 3.6.3)
##     textshape              1.7.3      2021-05-28 [1] CRAN (R 3.6.3)
##     tibble               * 3.1.6      2021-11-07 [1] CRAN (R 3.6.3)
##     tidyr                * 1.2.0      2022-02-01 [1] CRAN (R 3.6.3)
##     tidyselect             1.1.1      2021-04-30 [1] CRAN (R 3.6.3)
##     tidyverse            * 1.3.1      2021-04-15 [1] CRAN (R 3.6.3)
##     tzdb                   0.2.0      2021-10-27 [1] CRAN (R 3.6.3)
##     UpSetR                 1.4.0      2019-05-22 [1] CRAN (R 3.6.3)
##     usethis                2.1.6      2022-05-25 [2] CRAN (R 3.6.3)
##     utf8                   1.2.2      2021-07-24 [1] CRAN (R 3.6.3)
##     vctrs                  0.3.8      2021-04-29 [1] CRAN (R 3.6.3)
##     vegan                * 2.5-7      2020-11-28 [1] CRAN (R 3.6.3)
##     VennDiagram          * 1.7.1      2021-12-02 [1] CRAN (R 3.6.3)
##     viridisLite            0.4.0      2021-04-13 [2] CRAN (R 3.6.3)
##     vroom                  1.5.7      2021-11-30 [1] CRAN (R 3.6.3)
##     withr                  2.4.3      2021-11-30 [1] CRAN (R 3.6.3)
##     Wrench                 1.4.0      2019-10-29 [1] Bioconductor
##     xfun                   0.23       2021-05-15 [1] CRAN (R 3.6.3)
##     xlsx                 * 0.6.5      2020-11-10 [1] CRAN (R 3.6.3)
##     xlsxjars               0.6.1      2014-08-22 [1] CRAN (R 3.6.3)
##     XMAS                 * 0.0.0.9000 2022-03-23 [1] local
##     XML                    3.99-0.3   2020-01-20 [1] CRAN (R 3.6.3)
##     xml2                   1.3.3      2021-11-30 [2] CRAN (R 3.6.3)
##     xtable                 1.8-4      2019-04-21 [1] CRAN (R 3.6.3)
##     XVector                0.26.0     2019-10-29 [2] Bioconductor
##     xviz                 * 1.1.0      2021-01-14 [1] local
##     yaml                   2.2.2      2022-01-25 [1] CRAN (R 3.6.3)
##     zlibbioc               1.32.0     2019-10-29 [2] Bioconductor
## 
##  [1] /share/home/tongbangzhuo/R/x86_64-pc-linux-gnu-library/3.6
##  [2] /opt/R-3.6.3/lib/R/library
## 
##  V ── Loaded and on-disk version mismatch.
##  P ── Loaded and on-disk path mismatch.
##  R ── Package was removed from disk.
## 
## ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="preprocess-of-functional-analysis-of-kegg.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/XbiomeAnalysis/FunctionAnalysis/blob/main/01-MGS_functional_standard_analysis_demo.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Function standard analysis workflow.pdf", "Function standard analysis workflow.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
